// @arduino/arduino-iot-client/dist/model/ArduinoPropertytype.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(r, e) { var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (!t) { if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) { t && (r = t); var _n = 0, F = function F() {}; return { s: F, n: function n() { return _n >= r.length ? { done: !0 } : { done: !1, value: r[_n++] }; }, e: function e(r) { throw r; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var o, a = !0, u = !1; return { s: function s() { t = t.call(r); }, n: function n() { var r = t.next(); return a = r.done, r; }, e: function e(r) { u = !0, o = r; }, f: function f() { try { a || null == t["return"] || t["return"](); } finally { if (u) throw o; } } }; }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Arduino IoT Cloud API
 * Provides a set of endpoints to manage Arduino IoT Cloud **Devices**, **Things**, **Properties** and **Timeseries**. This API can be called just with any HTTP Client, or using one of these clients:  * [Javascript NPM package](https://www.npmjs.com/package/@arduino/arduino-iot-client)  * [Python PYPI Package](https://pypi.org/project/arduino-iot-client/)  * [Golang Module](https://github.com/arduino/iot-client-go)
 *
 * The version of the OpenAPI document: 2.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The ArduinoPropertytype model module.
 * @module model/ArduinoPropertytype
 * @version 3.0.0
 */
var ArduinoPropertytype = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ArduinoPropertytype</code>.
   * ArduinoPropertytype media type (default view)
   * @alias module:model/ArduinoPropertytype
   * @param declaration {String} The c++ type we are using for this variable type
   * @param deprecated {Boolean} Tell if this type is deprecated
   * @param name {String} The friendly name of the property type
   * @param rw {Boolean} Tell if the type allow a R/W permission
   * @param type {String} The api reference of this type
   */
  function ArduinoPropertytype(declaration, deprecated, name, rw, type) {
    _classCallCheck(this, ArduinoPropertytype);
    ArduinoPropertytype.initialize(this, declaration, deprecated, name, rw, type);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(ArduinoPropertytype, null, [{
    key: "initialize",
    value: function initialize(obj, declaration, deprecated, name, rw, type) {
      obj['declaration'] = declaration;
      obj['deprecated'] = deprecated;
      obj['name'] = name;
      obj['rw'] = rw;
      obj['type'] = type;
    }

    /**
     * Constructs a <code>ArduinoPropertytype</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ArduinoPropertytype} obj Optional instance to populate.
     * @return {module:model/ArduinoPropertytype} The populated <code>ArduinoPropertytype</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ArduinoPropertytype();
        if (data.hasOwnProperty('assistants')) {
          obj['assistants'] = _ApiClient["default"].convertToType(data['assistants'], ['String']);
        }
        if (data.hasOwnProperty('declaration')) {
          obj['declaration'] = _ApiClient["default"].convertToType(data['declaration'], 'String');
        }
        if (data.hasOwnProperty('deprecated')) {
          obj['deprecated'] = _ApiClient["default"].convertToType(data['deprecated'], 'Boolean');
        }
        if (data.hasOwnProperty('example')) {
          obj['example'] = _ApiClient["default"].convertToType(data['example'], 'String');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('rw')) {
          obj['rw'] = _ApiClient["default"].convertToType(data['rw'], 'Boolean');
        }
        if (data.hasOwnProperty('supersededBy')) {
          obj['supersededBy'] = _ApiClient["default"].convertToType(data['supersededBy'], 'String');
        }
        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
        if (data.hasOwnProperty('units')) {
          obj['units'] = _ApiClient["default"].convertToType(data['units'], ['String']);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>ArduinoPropertytype</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ArduinoPropertytype</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(ArduinoPropertytype.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
        // ensure the json data is an array
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      if (!Array.isArray(data['assistants'])) {
        throw new Error("Expected the field `assistants` to be an array in the JSON data but got " + data['assistants']);
      }
      // ensure the json data is a string
      if (data['declaration'] && !(typeof data['declaration'] === 'string' || data['declaration'] instanceof String)) {
        throw new Error("Expected the field `declaration` to be a primitive type in the JSON string but got " + data['declaration']);
      }
      // ensure the json data is a string
      if (data['example'] && !(typeof data['example'] === 'string' || data['example'] instanceof String)) {
        throw new Error("Expected the field `example` to be a primitive type in the JSON string but got " + data['example']);
      }
      // ensure the json data is a string
      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      }
      // ensure the json data is a string
      if (data['supersededBy'] && !(typeof data['supersededBy'] === 'string' || data['supersededBy'] instanceof String)) {
        throw new Error("Expected the field `supersededBy` to be a primitive type in the JSON string but got " + data['supersededBy']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['tags'])) {
        throw new Error("Expected the field `tags` to be an array in the JSON data but got " + data['tags']);
      }
      // ensure the json data is a string
      if (data['type'] && !(typeof data['type'] === 'string' || data['type'] instanceof String)) {
        throw new Error("Expected the field `type` to be a primitive type in the JSON string but got " + data['type']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['units'])) {
        throw new Error("Expected the field `units` to be an array in the JSON data but got " + data['units']);
      }
      return true;
    }
  }]);
}();
ArduinoPropertytype.RequiredProperties = ["declaration", "deprecated", "name", "rw", "type"];

/**
 * The voice assistants available for this type
 * @member {Array.<String>} assistants
 */
ArduinoPropertytype.prototype['assistants'] = undefined;

/**
 * The c++ type we are using for this variable type
 * @member {String} declaration
 */
ArduinoPropertytype.prototype['declaration'] = undefined;

/**
 * Tell if this type is deprecated
 * @member {Boolean} deprecated
 */
ArduinoPropertytype.prototype['deprecated'] = undefined;

/**
 * Example of use
 * @member {String} example
 */
ArduinoPropertytype.prototype['example'] = undefined;

/**
 * The friendly name of the property type
 * @member {String} name
 */
ArduinoPropertytype.prototype['name'] = undefined;

/**
 * Tell if the type allow a R/W permission
 * @member {Boolean} rw
 */
ArduinoPropertytype.prototype['rw'] = undefined;

/**
 * The type of property to use if it's deprecated
 * @member {String} supersededBy
 */
ArduinoPropertytype.prototype['supersededBy'] = undefined;

/**
 * The tags related to the type
 * @member {Array.<String>} tags
 */
ArduinoPropertytype.prototype['tags'] = undefined;

/**
 * The api reference of this type
 * @member {String} type
 */
ArduinoPropertytype.prototype['type'] = undefined;

/**
 * The measure units available for this type
 * @member {Array.<String>} units
 */
ArduinoPropertytype.prototype['units'] = undefined;
var _default = exports["default"] = ArduinoPropertytype;
