// @arduino/arduino-iot-client/dist/model/ArduinoTriggerTemplate.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ArduinoActionTemplate = _interopRequireDefault(require("./ArduinoActionTemplate"));
var _ArduinoLinkedDeviceTemplate = _interopRequireDefault(require("./ArduinoLinkedDeviceTemplate"));
var _ArduinoLinkedPropertyTemplate = _interopRequireDefault(require("./ArduinoLinkedPropertyTemplate"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(r, e) { var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (!t) { if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) { t && (r = t); var _n = 0, F = function F() {}; return { s: F, n: function n() { return _n >= r.length ? { done: !0 } : { done: !1, value: r[_n++] }; }, e: function e(r) { throw r; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var o, a = !0, u = !1; return { s: function s() { t = t.call(r); }, n: function n() { var r = t.next(); return a = r.done, r; }, e: function e(r) { u = !0, o = r; }, f: function f() { try { a || null == t["return"] || t["return"](); } finally { if (u) throw o; } } }; }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Arduino IoT Cloud API
 * Provides a set of endpoints to manage Arduino IoT Cloud **Devices**, **Things**, **Properties** and **Timeseries**. This API can be called just with any HTTP Client, or using one of these clients:  * [Javascript NPM package](https://www.npmjs.com/package/@arduino/arduino-iot-client)  * [Python PYPI Package](https://pypi.org/project/arduino-iot-client/)  * [Golang Module](https://github.com/arduino/iot-client-go)
 *
 * The version of the OpenAPI document: 2.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The ArduinoTriggerTemplate model module.
 * @module model/ArduinoTriggerTemplate
 * @version 3.0.0
 */
var ArduinoTriggerTemplate = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ArduinoTriggerTemplate</code>.
   * ArduinoTrigger_template media type (default view)
   * @alias module:model/ArduinoTriggerTemplate
   * @param id {String} The id of the trigger
   * @param name {String} The name of the trigger
   */
  function ArduinoTriggerTemplate(id, name) {
    _classCallCheck(this, ArduinoTriggerTemplate);
    ArduinoTriggerTemplate.initialize(this, id, name);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(ArduinoTriggerTemplate, null, [{
    key: "initialize",
    value: function initialize(obj, id, name) {
      obj['id'] = id;
      obj['name'] = name;
    }

    /**
     * Constructs a <code>ArduinoTriggerTemplate</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ArduinoTriggerTemplate} obj Optional instance to populate.
     * @return {module:model/ArduinoTriggerTemplate} The populated <code>ArduinoTriggerTemplate</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ArduinoTriggerTemplate();
        if (data.hasOwnProperty('actions')) {
          obj['actions'] = _ApiClient["default"].convertToType(data['actions'], [_ArduinoActionTemplate["default"]]);
        }
        if (data.hasOwnProperty('active')) {
          obj['active'] = _ApiClient["default"].convertToType(data['active'], 'Boolean');
        }
        if (data.hasOwnProperty('criteria')) {
          obj['criteria'] = _ApiClient["default"].convertToType(data['criteria'], 'String');
        }
        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }
        if (data.hasOwnProperty('grace_period_offline')) {
          obj['grace_period_offline'] = _ApiClient["default"].convertToType(data['grace_period_offline'], 'Number');
        }
        if (data.hasOwnProperty('grace_period_online')) {
          obj['grace_period_online'] = _ApiClient["default"].convertToType(data['grace_period_online'], 'Number');
        }
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
        if (data.hasOwnProperty('linked_devices')) {
          obj['linked_devices'] = _ApiClient["default"].convertToType(data['linked_devices'], [_ArduinoLinkedDeviceTemplate["default"]]);
        }
        if (data.hasOwnProperty('linked_property')) {
          obj['linked_property'] = _ArduinoLinkedPropertyTemplate["default"].constructFromObject(data['linked_property']);
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('organization_id')) {
          obj['organization_id'] = _ApiClient["default"].convertToType(data['organization_id'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>ArduinoTriggerTemplate</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ArduinoTriggerTemplate</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(ArduinoTriggerTemplate.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      if (data['actions']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['actions'])) {
          throw new Error("Expected the field `actions` to be an array in the JSON data but got " + data['actions']);
        }
        // validate the optional field `actions` (array)
        var _iterator2 = _createForOfIteratorHelper(data['actions']),
          _step2;
        try {
          for (_iterator2.s(); !(_step2 = _iterator2.n()).done;) {
            var item = _step2.value;
            _ArduinoActionTemplate["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator2.e(err);
        } finally {
          _iterator2.f();
        }
        ;
      }
      // ensure the json data is a string
      if (data['criteria'] && !(typeof data['criteria'] === 'string' || data['criteria'] instanceof String)) {
        throw new Error("Expected the field `criteria` to be a primitive type in the JSON string but got " + data['criteria']);
      }
      // ensure the json data is a string
      if (data['description'] && !(typeof data['description'] === 'string' || data['description'] instanceof String)) {
        throw new Error("Expected the field `description` to be a primitive type in the JSON string but got " + data['description']);
      }
      // ensure the json data is a string
      if (data['id'] && !(typeof data['id'] === 'string' || data['id'] instanceof String)) {
        throw new Error("Expected the field `id` to be a primitive type in the JSON string but got " + data['id']);
      }
      if (data['linked_devices']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['linked_devices'])) {
          throw new Error("Expected the field `linked_devices` to be an array in the JSON data but got " + data['linked_devices']);
        }
        // validate the optional field `linked_devices` (array)
        var _iterator3 = _createForOfIteratorHelper(data['linked_devices']),
          _step3;
        try {
          for (_iterator3.s(); !(_step3 = _iterator3.n()).done;) {
            var _item = _step3.value;
            _ArduinoLinkedDeviceTemplate["default"].validateJSON(_item);
          }
        } catch (err) {
          _iterator3.e(err);
        } finally {
          _iterator3.f();
        }
        ;
      }
      // validate the optional field `linked_property`
      if (data['linked_property']) {
        // data not null
        _ArduinoLinkedPropertyTemplate["default"].validateJSON(data['linked_property']);
      }
      // ensure the json data is a string
      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      }
      // ensure the json data is a string
      if (data['organization_id'] && !(typeof data['organization_id'] === 'string' || data['organization_id'] instanceof String)) {
        throw new Error("Expected the field `organization_id` to be a primitive type in the JSON string but got " + data['organization_id']);
      }
      return true;
    }
  }]);
}();
ArduinoTriggerTemplate.RequiredProperties = ["id", "name"];

/**
 * A list of actions associated with the trigger
 * @member {Array.<module:model/ArduinoActionTemplate>} actions
 */
ArduinoTriggerTemplate.prototype['actions'] = undefined;

/**
 * Is true if the trigger is enabled
 * @member {Boolean} active
 */
ArduinoTriggerTemplate.prototype['active'] = undefined;

/**
 * The criteria of the trigger, could be INCLUDE or EXCLUDE
 * @member {String} criteria
 */
ArduinoTriggerTemplate.prototype['criteria'] = undefined;

/**
 * The description of the trigger
 * @member {String} description
 */
ArduinoTriggerTemplate.prototype['description'] = undefined;

/**
 * The amount of seconds the trigger will wait before considering a matching device as offline
 * @member {Number} grace_period_offline
 */
ArduinoTriggerTemplate.prototype['grace_period_offline'] = undefined;

/**
 * The amount of seconds the trigger will wait before considering a matching device as online
 * @member {Number} grace_period_online
 */
ArduinoTriggerTemplate.prototype['grace_period_online'] = undefined;

/**
 * The id of the trigger
 * @member {String} id
 */
ArduinoTriggerTemplate.prototype['id'] = undefined;

/**
 * A list of devices the trigger is associated to
 * @member {Array.<module:model/ArduinoLinkedDeviceTemplate>} linked_devices
 */
ArduinoTriggerTemplate.prototype['linked_devices'] = undefined;

/**
 * @member {module:model/ArduinoLinkedPropertyTemplate} linked_property
 */
ArduinoTriggerTemplate.prototype['linked_property'] = undefined;

/**
 * The name of the trigger
 * @member {String} name
 */
ArduinoTriggerTemplate.prototype['name'] = undefined;

/**
 * Id of the organization the trigger belongs to
 * @member {String} organization_id
 */
ArduinoTriggerTemplate.prototype['organization_id'] = undefined;
var _default = exports["default"] = ArduinoTriggerTemplate;
