// @arduino/arduino-iot-client/dist/model/ArduinoDevicev2.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ArduinoDevicev2SimpleProperties = _interopRequireDefault(require("./ArduinoDevicev2SimpleProperties"));
var _ArduinoDevicev2Webhook = _interopRequireDefault(require("./ArduinoDevicev2Webhook"));
var _ArduinoThing = _interopRequireDefault(require("./ArduinoThing"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(r, e) { var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (!t) { if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) { t && (r = t); var _n = 0, F = function F() {}; return { s: F, n: function n() { return _n >= r.length ? { done: !0 } : { done: !1, value: r[_n++] }; }, e: function e(r) { throw r; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var o, a = !0, u = !1; return { s: function s() { t = t.call(r); }, n: function n() { var r = t.next(); return a = r.done, r; }, e: function e(r) { u = !0, o = r; }, f: function f() { try { a || null == t["return"] || t["return"](); } finally { if (u) throw o; } } }; }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Arduino IoT Cloud API
 * Provides a set of endpoints to manage Arduino IoT Cloud **Devices**, **Things**, **Properties** and **Timeseries**. This API can be called just with any HTTP Client, or using one of these clients:  * [Javascript NPM package](https://www.npmjs.com/package/@arduino/arduino-iot-client)  * [Python PYPI Package](https://pypi.org/project/arduino-iot-client/)  * [Golang Module](https://github.com/arduino/iot-client-go)
 *
 * The version of the OpenAPI document: 2.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The ArduinoDevicev2 model module.
 * @module model/ArduinoDevicev2
 * @version 3.0.0
 */
var ArduinoDevicev2 = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ArduinoDevicev2</code>.
   * ArduinoDevicev2 media type (default view)
   * @alias module:model/ArduinoDevicev2
   * @param href {String} The api reference of this device
   * @param id {String} The arn of the device
   * @param label {String} The label of the device
   * @param name {String} The friendly name of the device
   * @param serial {String} The serial uuid of the device
   * @param type {String} The type of the device
   * @param userId {String} The id of the user
   */
  function ArduinoDevicev2(href, id, label, name, serial, type, userId) {
    _classCallCheck(this, ArduinoDevicev2);
    ArduinoDevicev2.initialize(this, href, id, label, name, serial, type, userId);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(ArduinoDevicev2, null, [{
    key: "initialize",
    value: function initialize(obj, href, id, label, name, serial, type, userId) {
      obj['href'] = href;
      obj['id'] = id;
      obj['label'] = label;
      obj['name'] = name;
      obj['serial'] = serial;
      obj['type'] = type;
      obj['user_id'] = userId;
    }

    /**
     * Constructs a <code>ArduinoDevicev2</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ArduinoDevicev2} obj Optional instance to populate.
     * @return {module:model/ArduinoDevicev2} The populated <code>ArduinoDevicev2</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ArduinoDevicev2();
        if (data.hasOwnProperty('connection_type')) {
          obj['connection_type'] = _ApiClient["default"].convertToType(data['connection_type'], 'String');
        }
        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Date');
        }
        if (data.hasOwnProperty('deleted_at')) {
          obj['deleted_at'] = _ApiClient["default"].convertToType(data['deleted_at'], 'Date');
        }
        if (data.hasOwnProperty('device_status')) {
          obj['device_status'] = _ApiClient["default"].convertToType(data['device_status'], 'String');
        }
        if (data.hasOwnProperty('events')) {
          obj['events'] = _ApiClient["default"].convertToType(data['events'], [_ArduinoDevicev2SimpleProperties["default"]]);
        }
        if (data.hasOwnProperty('fqbn')) {
          obj['fqbn'] = _ApiClient["default"].convertToType(data['fqbn'], 'String');
        }
        if (data.hasOwnProperty('href')) {
          obj['href'] = _ApiClient["default"].convertToType(data['href'], 'String');
        }
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
        if (data.hasOwnProperty('label')) {
          obj['label'] = _ApiClient["default"].convertToType(data['label'], 'String');
        }
        if (data.hasOwnProperty('last_activity_at')) {
          obj['last_activity_at'] = _ApiClient["default"].convertToType(data['last_activity_at'], 'Date');
        }
        if (data.hasOwnProperty('latest_wifi_fw_version')) {
          obj['latest_wifi_fw_version'] = _ApiClient["default"].convertToType(data['latest_wifi_fw_version'], 'String');
        }
        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], {
            'String': Object
          });
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('no_sketch')) {
          obj['no_sketch'] = _ApiClient["default"].convertToType(data['no_sketch'], 'Boolean');
        }
        if (data.hasOwnProperty('organization_id')) {
          obj['organization_id'] = _ApiClient["default"].convertToType(data['organization_id'], 'String');
        }
        if (data.hasOwnProperty('ota_available')) {
          obj['ota_available'] = _ApiClient["default"].convertToType(data['ota_available'], 'Boolean');
        }
        if (data.hasOwnProperty('ota_compatible')) {
          obj['ota_compatible'] = _ApiClient["default"].convertToType(data['ota_compatible'], 'Boolean');
        }
        if (data.hasOwnProperty('required_wifi_fw_version')) {
          obj['required_wifi_fw_version'] = _ApiClient["default"].convertToType(data['required_wifi_fw_version'], 'String');
        }
        if (data.hasOwnProperty('serial')) {
          obj['serial'] = _ApiClient["default"].convertToType(data['serial'], 'String');
        }
        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], {
            'String': Object
          });
        }
        if (data.hasOwnProperty('thing')) {
          obj['thing'] = _ArduinoThing["default"].constructFromObject(data['thing']);
        }
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
        if (data.hasOwnProperty('updated_at')) {
          obj['updated_at'] = _ApiClient["default"].convertToType(data['updated_at'], 'Date');
        }
        if (data.hasOwnProperty('user_id')) {
          obj['user_id'] = _ApiClient["default"].convertToType(data['user_id'], 'String');
        }
        if (data.hasOwnProperty('webhooks')) {
          obj['webhooks'] = _ApiClient["default"].convertToType(data['webhooks'], [_ArduinoDevicev2Webhook["default"]]);
        }
        if (data.hasOwnProperty('wifi_fw_version')) {
          obj['wifi_fw_version'] = _ApiClient["default"].convertToType(data['wifi_fw_version'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>ArduinoDevicev2</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ArduinoDevicev2</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(ArduinoDevicev2.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
        // ensure the json data is a string
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      if (data['connection_type'] && !(typeof data['connection_type'] === 'string' || data['connection_type'] instanceof String)) {
        throw new Error("Expected the field `connection_type` to be a primitive type in the JSON string but got " + data['connection_type']);
      }
      // ensure the json data is a string
      if (data['device_status'] && !(typeof data['device_status'] === 'string' || data['device_status'] instanceof String)) {
        throw new Error("Expected the field `device_status` to be a primitive type in the JSON string but got " + data['device_status']);
      }
      if (data['events']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['events'])) {
          throw new Error("Expected the field `events` to be an array in the JSON data but got " + data['events']);
        }
        // validate the optional field `events` (array)
        var _iterator2 = _createForOfIteratorHelper(data['events']),
          _step2;
        try {
          for (_iterator2.s(); !(_step2 = _iterator2.n()).done;) {
            var item = _step2.value;
            _ArduinoDevicev2SimpleProperties["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator2.e(err);
        } finally {
          _iterator2.f();
        }
        ;
      }
      // ensure the json data is a string
      if (data['fqbn'] && !(typeof data['fqbn'] === 'string' || data['fqbn'] instanceof String)) {
        throw new Error("Expected the field `fqbn` to be a primitive type in the JSON string but got " + data['fqbn']);
      }
      // ensure the json data is a string
      if (data['href'] && !(typeof data['href'] === 'string' || data['href'] instanceof String)) {
        throw new Error("Expected the field `href` to be a primitive type in the JSON string but got " + data['href']);
      }
      // ensure the json data is a string
      if (data['id'] && !(typeof data['id'] === 'string' || data['id'] instanceof String)) {
        throw new Error("Expected the field `id` to be a primitive type in the JSON string but got " + data['id']);
      }
      // ensure the json data is a string
      if (data['label'] && !(typeof data['label'] === 'string' || data['label'] instanceof String)) {
        throw new Error("Expected the field `label` to be a primitive type in the JSON string but got " + data['label']);
      }
      // ensure the json data is a string
      if (data['latest_wifi_fw_version'] && !(typeof data['latest_wifi_fw_version'] === 'string' || data['latest_wifi_fw_version'] instanceof String)) {
        throw new Error("Expected the field `latest_wifi_fw_version` to be a primitive type in the JSON string but got " + data['latest_wifi_fw_version']);
      }
      // ensure the json data is a string
      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      }
      // ensure the json data is a string
      if (data['organization_id'] && !(typeof data['organization_id'] === 'string' || data['organization_id'] instanceof String)) {
        throw new Error("Expected the field `organization_id` to be a primitive type in the JSON string but got " + data['organization_id']);
      }
      // ensure the json data is a string
      if (data['required_wifi_fw_version'] && !(typeof data['required_wifi_fw_version'] === 'string' || data['required_wifi_fw_version'] instanceof String)) {
        throw new Error("Expected the field `required_wifi_fw_version` to be a primitive type in the JSON string but got " + data['required_wifi_fw_version']);
      }
      // ensure the json data is a string
      if (data['serial'] && !(typeof data['serial'] === 'string' || data['serial'] instanceof String)) {
        throw new Error("Expected the field `serial` to be a primitive type in the JSON string but got " + data['serial']);
      }
      // validate the optional field `thing`
      if (data['thing']) {
        // data not null
        _ArduinoThing["default"].validateJSON(data['thing']);
      }
      // ensure the json data is a string
      if (data['type'] && !(typeof data['type'] === 'string' || data['type'] instanceof String)) {
        throw new Error("Expected the field `type` to be a primitive type in the JSON string but got " + data['type']);
      }
      // ensure the json data is a string
      if (data['user_id'] && !(typeof data['user_id'] === 'string' || data['user_id'] instanceof String)) {
        throw new Error("Expected the field `user_id` to be a primitive type in the JSON string but got " + data['user_id']);
      }
      if (data['webhooks']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['webhooks'])) {
          throw new Error("Expected the field `webhooks` to be an array in the JSON data but got " + data['webhooks']);
        }
        // validate the optional field `webhooks` (array)
        var _iterator3 = _createForOfIteratorHelper(data['webhooks']),
          _step3;
        try {
          for (_iterator3.s(); !(_step3 = _iterator3.n()).done;) {
            var _item = _step3.value;
            _ArduinoDevicev2Webhook["default"].validateJSON(_item);
          }
        } catch (err) {
          _iterator3.e(err);
        } finally {
          _iterator3.f();
        }
        ;
      }
      // ensure the json data is a string
      if (data['wifi_fw_version'] && !(typeof data['wifi_fw_version'] === 'string' || data['wifi_fw_version'] instanceof String)) {
        throw new Error("Expected the field `wifi_fw_version` to be a primitive type in the JSON string but got " + data['wifi_fw_version']);
      }
      return true;
    }
  }]);
}();
ArduinoDevicev2.RequiredProperties = ["href", "id", "label", "name", "serial", "type", "user_id"];

/**
 * The type of the connections selected by the user when multiple connections are available
 * @member {module:model/ArduinoDevicev2.ConnectionTypeEnum} connection_type
 */
ArduinoDevicev2.prototype['connection_type'] = undefined;

/**
 * Creation date of the device
 * @member {Date} created_at
 */
ArduinoDevicev2.prototype['created_at'] = undefined;

/**
 * Deletion date of the trigger
 * @member {Date} deleted_at
 */
ArduinoDevicev2.prototype['deleted_at'] = undefined;

/**
 * The connection status of the device
 * @member {module:model/ArduinoDevicev2.DeviceStatusEnum} device_status
 */
ArduinoDevicev2.prototype['device_status'] = undefined;

/**
 * ArduinoDevicev2SimplePropertiesCollection is the media type for an array of ArduinoDevicev2SimpleProperties (default view)
 * @member {Array.<module:model/ArduinoDevicev2SimpleProperties>} events
 */
ArduinoDevicev2.prototype['events'] = undefined;

/**
 * The fully qualified board name
 * @member {String} fqbn
 */
ArduinoDevicev2.prototype['fqbn'] = undefined;

/**
 * The api reference of this device
 * @member {String} href
 */
ArduinoDevicev2.prototype['href'] = undefined;

/**
 * The arn of the device
 * @member {String} id
 */
ArduinoDevicev2.prototype['id'] = undefined;

/**
 * The label of the device
 * @member {String} label
 */
ArduinoDevicev2.prototype['label'] = undefined;

/**
 * Last activity date
 * @member {Date} last_activity_at
 */
ArduinoDevicev2.prototype['last_activity_at'] = undefined;

/**
 * The latest version of the NINA/WIFI101 firmware available for this device
 * @member {String} latest_wifi_fw_version
 */
ArduinoDevicev2.prototype['latest_wifi_fw_version'] = undefined;

/**
 * The metadata of the device
 * @member {Object.<String, Object>} metadata
 */
ArduinoDevicev2.prototype['metadata'] = undefined;

/**
 * The friendly name of the device
 * @member {String} name
 */
ArduinoDevicev2.prototype['name'] = undefined;

/**
 * True if the device type can not have an associated sketch
 * @member {Boolean} no_sketch
 */
ArduinoDevicev2.prototype['no_sketch'] = undefined;

/**
 * Id of the organization the device belongs to
 * @member {String} organization_id
 */
ArduinoDevicev2.prototype['organization_id'] = undefined;

/**
 * True if the device type is ready to receive OTA updated
 * @member {Boolean} ota_available
 */
ArduinoDevicev2.prototype['ota_available'] = undefined;

/**
 * True if the device type is OTA compatible
 * @member {Boolean} ota_compatible
 */
ArduinoDevicev2.prototype['ota_compatible'] = undefined;

/**
 * The required version of the NINA/WIFI101 firmware needed by IoT Cloud
 * @member {String} required_wifi_fw_version
 */
ArduinoDevicev2.prototype['required_wifi_fw_version'] = undefined;

/**
 * The serial uuid of the device
 * @member {String} serial
 */
ArduinoDevicev2.prototype['serial'] = undefined;

/**
 * Tags belonging to the device
 * @member {Object.<String, Object>} tags
 */
ArduinoDevicev2.prototype['tags'] = undefined;

/**
 * @member {module:model/ArduinoThing} thing
 */
ArduinoDevicev2.prototype['thing'] = undefined;

/**
 * The type of the device
 * @member {String} type
 */
ArduinoDevicev2.prototype['type'] = undefined;

/**
 * Update date of the trigger
 * @member {Date} updated_at
 */
ArduinoDevicev2.prototype['updated_at'] = undefined;

/**
 * The id of the user
 * @member {String} user_id
 */
ArduinoDevicev2.prototype['user_id'] = undefined;

/**
 * ArduinoDevicev2WebhookCollection is the media type for an array of ArduinoDevicev2Webhook (default view)
 * @member {Array.<module:model/ArduinoDevicev2Webhook>} webhooks
 */
ArduinoDevicev2.prototype['webhooks'] = undefined;

/**
 * The version of the NINA/WIFI101 firmware running on the device
 * @member {String} wifi_fw_version
 */
ArduinoDevicev2.prototype['wifi_fw_version'] = undefined;

/**
 * Allowed values for the <code>connection_type</code> property.
 * @enum {String}
 * @readonly
 */
ArduinoDevicev2['ConnectionTypeEnum'] = {
  /**
   * value: "wifi"
   * @const
   */
  "wifi": "wifi",
  /**
   * value: "eth"
   * @const
   */
  "eth": "eth",
  /**
   * value: "wifiandsecret"
   * @const
   */
  "wifiandsecret": "wifiandsecret",
  /**
   * value: "gsm"
   * @const
   */
  "gsm": "gsm",
  /**
   * value: "nb"
   * @const
   */
  "nb": "nb",
  /**
   * value: "lora"
   * @const
   */
  "lora": "lora",
  /**
   * value: "catm1"
   * @const
   */
  "catm1": "catm1",
  /**
   * value: "cellular"
   * @const
   */
  "cellular": "cellular"
};

/**
 * Allowed values for the <code>device_status</code> property.
 * @enum {String}
 * @readonly
 */
ArduinoDevicev2['DeviceStatusEnum'] = {
  /**
   * value: "ONLINE"
   * @const
   */
  "ONLINE": "ONLINE",
  /**
   * value: "OFFLINE"
   * @const
   */
  "OFFLINE": "OFFLINE",
  /**
   * value: "UNKNOWN"
   * @const
   */
  "UNKNOWN": "UNKNOWN"
};
var _default = exports["default"] = ArduinoDevicev2;
