// @arduino/arduino-iot-client/dist/model/ArduinoCompressedv2.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(r, e) { var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (!t) { if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) { t && (r = t); var _n = 0, F = function F() {}; return { s: F, n: function n() { return _n >= r.length ? { done: !0 } : { done: !1, value: r[_n++] }; }, e: function e(r) { throw r; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var o, a = !0, u = !1; return { s: function s() { t = t.call(r); }, n: function n() { var r = t.next(); return a = r.done, r; }, e: function e(r) { u = !0, o = r; }, f: function f() { try { a || null == t["return"] || t["return"](); } finally { if (u) throw o; } } }; }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Arduino IoT Cloud API
 * Provides a set of endpoints to manage Arduino IoT Cloud **Devices**, **Things**, **Properties** and **Timeseries**. This API can be called just with any HTTP Client, or using one of these clients:  * [Javascript NPM package](https://www.npmjs.com/package/@arduino/arduino-iot-client)  * [Python PYPI Package](https://pypi.org/project/arduino-iot-client/)  * [Golang Module](https://github.com/arduino/iot-client-go)
 *
 * The version of the OpenAPI document: 2.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The ArduinoCompressedv2 model module.
 * @module model/ArduinoCompressedv2
 * @version 3.0.0
 */
var ArduinoCompressedv2 = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ArduinoCompressedv2</code>.
   * compressed contains the info from which to generate the certificate (default view)
   * @alias module:model/ArduinoCompressedv2
   * @param notAfter {Date} The ending date of the certificate
   * @param notBefore {Date} The starting date of the certificate
   * @param serial {String} The serial number of the certificate
   * @param signature {String} The signature of the certificate
   * @param signatureAsn1X {String} The ASN1 X component of certificate signature
   * @param signatureAsn1Y {String} The ASN1 Y component of certificate signature
   */
  function ArduinoCompressedv2(notAfter, notBefore, serial, signature, signatureAsn1X, signatureAsn1Y) {
    _classCallCheck(this, ArduinoCompressedv2);
    ArduinoCompressedv2.initialize(this, notAfter, notBefore, serial, signature, signatureAsn1X, signatureAsn1Y);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(ArduinoCompressedv2, null, [{
    key: "initialize",
    value: function initialize(obj, notAfter, notBefore, serial, signature, signatureAsn1X, signatureAsn1Y) {
      obj['not_after'] = notAfter;
      obj['not_before'] = notBefore;
      obj['serial'] = serial;
      obj['signature'] = signature;
      obj['signature_asn1_x'] = signatureAsn1X;
      obj['signature_asn1_y'] = signatureAsn1Y;
    }

    /**
     * Constructs a <code>ArduinoCompressedv2</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ArduinoCompressedv2} obj Optional instance to populate.
     * @return {module:model/ArduinoCompressedv2} The populated <code>ArduinoCompressedv2</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ArduinoCompressedv2();
        if (data.hasOwnProperty('authority_key_identifier')) {
          obj['authority_key_identifier'] = _ApiClient["default"].convertToType(data['authority_key_identifier'], 'String');
        }
        if (data.hasOwnProperty('not_after')) {
          obj['not_after'] = _ApiClient["default"].convertToType(data['not_after'], 'Date');
        }
        if (data.hasOwnProperty('not_before')) {
          obj['not_before'] = _ApiClient["default"].convertToType(data['not_before'], 'Date');
        }
        if (data.hasOwnProperty('serial')) {
          obj['serial'] = _ApiClient["default"].convertToType(data['serial'], 'String');
        }
        if (data.hasOwnProperty('signature')) {
          obj['signature'] = _ApiClient["default"].convertToType(data['signature'], 'String');
        }
        if (data.hasOwnProperty('signature_asn1_x')) {
          obj['signature_asn1_x'] = _ApiClient["default"].convertToType(data['signature_asn1_x'], 'String');
        }
        if (data.hasOwnProperty('signature_asn1_y')) {
          obj['signature_asn1_y'] = _ApiClient["default"].convertToType(data['signature_asn1_y'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>ArduinoCompressedv2</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ArduinoCompressedv2</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(ArduinoCompressedv2.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
        // ensure the json data is a string
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      if (data['authority_key_identifier'] && !(typeof data['authority_key_identifier'] === 'string' || data['authority_key_identifier'] instanceof String)) {
        throw new Error("Expected the field `authority_key_identifier` to be a primitive type in the JSON string but got " + data['authority_key_identifier']);
      }
      // ensure the json data is a string
      if (data['serial'] && !(typeof data['serial'] === 'string' || data['serial'] instanceof String)) {
        throw new Error("Expected the field `serial` to be a primitive type in the JSON string but got " + data['serial']);
      }
      // ensure the json data is a string
      if (data['signature'] && !(typeof data['signature'] === 'string' || data['signature'] instanceof String)) {
        throw new Error("Expected the field `signature` to be a primitive type in the JSON string but got " + data['signature']);
      }
      // ensure the json data is a string
      if (data['signature_asn1_x'] && !(typeof data['signature_asn1_x'] === 'string' || data['signature_asn1_x'] instanceof String)) {
        throw new Error("Expected the field `signature_asn1_x` to be a primitive type in the JSON string but got " + data['signature_asn1_x']);
      }
      // ensure the json data is a string
      if (data['signature_asn1_y'] && !(typeof data['signature_asn1_y'] === 'string' || data['signature_asn1_y'] instanceof String)) {
        throw new Error("Expected the field `signature_asn1_y` to be a primitive type in the JSON string but got " + data['signature_asn1_y']);
      }
      return true;
    }
  }]);
}();
ArduinoCompressedv2.RequiredProperties = ["not_after", "not_before", "serial", "signature", "signature_asn1_x", "signature_asn1_y"];

/**
 * The Authority Key Identifier of the certificate
 * @member {String} authority_key_identifier
 */
ArduinoCompressedv2.prototype['authority_key_identifier'] = undefined;

/**
 * The ending date of the certificate
 * @member {Date} not_after
 */
ArduinoCompressedv2.prototype['not_after'] = undefined;

/**
 * The starting date of the certificate
 * @member {Date} not_before
 */
ArduinoCompressedv2.prototype['not_before'] = undefined;

/**
 * The serial number of the certificate
 * @member {String} serial
 */
ArduinoCompressedv2.prototype['serial'] = undefined;

/**
 * The signature of the certificate
 * @member {String} signature
 */
ArduinoCompressedv2.prototype['signature'] = undefined;

/**
 * The ASN1 X component of certificate signature
 * @member {String} signature_asn1_x
 */
ArduinoCompressedv2.prototype['signature_asn1_x'] = undefined;

/**
 * The ASN1 Y component of certificate signature
 * @member {String} signature_asn1_y
 */
ArduinoCompressedv2.prototype['signature_asn1_y'] = undefined;
var _default = exports["default"] = ArduinoCompressedv2;
