// @arduino/arduino-iot-client/dist/model/ArduinoThing.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ArduinoProperty = _interopRequireDefault(require("./ArduinoProperty"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(r, e) { var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (!t) { if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) { t && (r = t); var _n = 0, F = function F() {}; return { s: F, n: function n() { return _n >= r.length ? { done: !0 } : { done: !1, value: r[_n++] }; }, e: function e(r) { throw r; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var o, a = !0, u = !1; return { s: function s() { t = t.call(r); }, n: function n() { var r = t.next(); return a = r.done, r; }, e: function e(r) { u = !0, o = r; }, f: function f() { try { a || null == t["return"] || t["return"](); } finally { if (u) throw o; } } }; }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Arduino IoT Cloud API
 * Provides a set of endpoints to manage Arduino IoT Cloud **Devices**, **Things**, **Properties** and **Timeseries**. This API can be called just with any HTTP Client, or using one of these clients:  * [Javascript NPM package](https://www.npmjs.com/package/@arduino/arduino-iot-client)  * [Python PYPI Package](https://pypi.org/project/arduino-iot-client/)  * [Golang Module](https://github.com/arduino/iot-client-go)
 *
 * The version of the OpenAPI document: 2.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The ArduinoThing model module.
 * @module model/ArduinoThing
 * @version 3.0.0
 */
var ArduinoThing = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ArduinoThing</code>.
   * ArduinoThing media type (default view)
   * @alias module:model/ArduinoThing
   * @param href {String} The api reference of this thing
   * @param id {String} The id of the thing
   * @param name {String} The friendly name of the thing
   * @param timezone {String} Time zone of the thing
   * @param userId {String} The user id of the owner
   */
  function ArduinoThing(href, id, name, timezone, userId) {
    _classCallCheck(this, ArduinoThing);
    ArduinoThing.initialize(this, href, id, name, timezone, userId);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(ArduinoThing, null, [{
    key: "initialize",
    value: function initialize(obj, href, id, name, timezone, userId) {
      obj['href'] = href;
      obj['id'] = id;
      obj['name'] = name;
      obj['timezone'] = timezone;
      obj['user_id'] = userId;
    }

    /**
     * Constructs a <code>ArduinoThing</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ArduinoThing} obj Optional instance to populate.
     * @return {module:model/ArduinoThing} The populated <code>ArduinoThing</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ArduinoThing();
        if (data.hasOwnProperty('assistant')) {
          obj['assistant'] = _ApiClient["default"].convertToType(data['assistant'], 'String');
        }
        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Date');
        }
        if (data.hasOwnProperty('deleted_at')) {
          obj['deleted_at'] = _ApiClient["default"].convertToType(data['deleted_at'], 'Date');
        }
        if (data.hasOwnProperty('device_fqbn')) {
          obj['device_fqbn'] = _ApiClient["default"].convertToType(data['device_fqbn'], 'String');
        }
        if (data.hasOwnProperty('device_id')) {
          obj['device_id'] = _ApiClient["default"].convertToType(data['device_id'], 'String');
        }
        if (data.hasOwnProperty('device_name')) {
          obj['device_name'] = _ApiClient["default"].convertToType(data['device_name'], 'String');
        }
        if (data.hasOwnProperty('device_type')) {
          obj['device_type'] = _ApiClient["default"].convertToType(data['device_type'], 'String');
        }
        if (data.hasOwnProperty('href')) {
          obj['href'] = _ApiClient["default"].convertToType(data['href'], 'String');
        }
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('organization_id')) {
          obj['organization_id'] = _ApiClient["default"].convertToType(data['organization_id'], 'String');
        }
        if (data.hasOwnProperty('properties')) {
          obj['properties'] = _ApiClient["default"].convertToType(data['properties'], [_ArduinoProperty["default"]]);
        }
        if (data.hasOwnProperty('properties_count')) {
          obj['properties_count'] = _ApiClient["default"].convertToType(data['properties_count'], 'Number');
        }
        if (data.hasOwnProperty('sketch_id')) {
          obj['sketch_id'] = _ApiClient["default"].convertToType(data['sketch_id'], 'String');
        }
        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], {
            'String': Object
          });
        }
        if (data.hasOwnProperty('timezone')) {
          obj['timezone'] = _ApiClient["default"].convertToType(data['timezone'], 'String');
        }
        if (data.hasOwnProperty('updated_at')) {
          obj['updated_at'] = _ApiClient["default"].convertToType(data['updated_at'], 'Date');
        }
        if (data.hasOwnProperty('user_id')) {
          obj['user_id'] = _ApiClient["default"].convertToType(data['user_id'], 'String');
        }
        if (data.hasOwnProperty('webhook_active')) {
          obj['webhook_active'] = _ApiClient["default"].convertToType(data['webhook_active'], 'Boolean');
        }
        if (data.hasOwnProperty('webhook_uri')) {
          obj['webhook_uri'] = _ApiClient["default"].convertToType(data['webhook_uri'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>ArduinoThing</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ArduinoThing</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(ArduinoThing.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
        // ensure the json data is a string
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      if (data['assistant'] && !(typeof data['assistant'] === 'string' || data['assistant'] instanceof String)) {
        throw new Error("Expected the field `assistant` to be a primitive type in the JSON string but got " + data['assistant']);
      }
      // ensure the json data is a string
      if (data['device_fqbn'] && !(typeof data['device_fqbn'] === 'string' || data['device_fqbn'] instanceof String)) {
        throw new Error("Expected the field `device_fqbn` to be a primitive type in the JSON string but got " + data['device_fqbn']);
      }
      // ensure the json data is a string
      if (data['device_id'] && !(typeof data['device_id'] === 'string' || data['device_id'] instanceof String)) {
        throw new Error("Expected the field `device_id` to be a primitive type in the JSON string but got " + data['device_id']);
      }
      // ensure the json data is a string
      if (data['device_name'] && !(typeof data['device_name'] === 'string' || data['device_name'] instanceof String)) {
        throw new Error("Expected the field `device_name` to be a primitive type in the JSON string but got " + data['device_name']);
      }
      // ensure the json data is a string
      if (data['device_type'] && !(typeof data['device_type'] === 'string' || data['device_type'] instanceof String)) {
        throw new Error("Expected the field `device_type` to be a primitive type in the JSON string but got " + data['device_type']);
      }
      // ensure the json data is a string
      if (data['href'] && !(typeof data['href'] === 'string' || data['href'] instanceof String)) {
        throw new Error("Expected the field `href` to be a primitive type in the JSON string but got " + data['href']);
      }
      // ensure the json data is a string
      if (data['id'] && !(typeof data['id'] === 'string' || data['id'] instanceof String)) {
        throw new Error("Expected the field `id` to be a primitive type in the JSON string but got " + data['id']);
      }
      // ensure the json data is a string
      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      }
      // ensure the json data is a string
      if (data['organization_id'] && !(typeof data['organization_id'] === 'string' || data['organization_id'] instanceof String)) {
        throw new Error("Expected the field `organization_id` to be a primitive type in the JSON string but got " + data['organization_id']);
      }
      if (data['properties']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['properties'])) {
          throw new Error("Expected the field `properties` to be an array in the JSON data but got " + data['properties']);
        }
        // validate the optional field `properties` (array)
        var _iterator2 = _createForOfIteratorHelper(data['properties']),
          _step2;
        try {
          for (_iterator2.s(); !(_step2 = _iterator2.n()).done;) {
            var item = _step2.value;
            _ArduinoProperty["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator2.e(err);
        } finally {
          _iterator2.f();
        }
        ;
      }
      // ensure the json data is a string
      if (data['sketch_id'] && !(typeof data['sketch_id'] === 'string' || data['sketch_id'] instanceof String)) {
        throw new Error("Expected the field `sketch_id` to be a primitive type in the JSON string but got " + data['sketch_id']);
      }
      // ensure the json data is a string
      if (data['timezone'] && !(typeof data['timezone'] === 'string' || data['timezone'] instanceof String)) {
        throw new Error("Expected the field `timezone` to be a primitive type in the JSON string but got " + data['timezone']);
      }
      // ensure the json data is a string
      if (data['user_id'] && !(typeof data['user_id'] === 'string' || data['user_id'] instanceof String)) {
        throw new Error("Expected the field `user_id` to be a primitive type in the JSON string but got " + data['user_id']);
      }
      // ensure the json data is a string
      if (data['webhook_uri'] && !(typeof data['webhook_uri'] === 'string' || data['webhook_uri'] instanceof String)) {
        throw new Error("Expected the field `webhook_uri` to be a primitive type in the JSON string but got " + data['webhook_uri']);
      }
      return true;
    }
  }]);
}();
ArduinoThing.RequiredProperties = ["href", "id", "name", "timezone", "user_id"];

/**
 * The kind of voice assistant the thing is connected to, it can be ALEXA | GOOGLE | NONE
 * @member {module:model/ArduinoThing.AssistantEnum} assistant
 */
ArduinoThing.prototype['assistant'] = undefined;

/**
 * Creation date of the thing
 * @member {Date} created_at
 */
ArduinoThing.prototype['created_at'] = undefined;

/**
 * Delete date of the thing
 * @member {Date} deleted_at
 */
ArduinoThing.prototype['deleted_at'] = undefined;

/**
 * The fqbn of the attached device, if any
 * @member {String} device_fqbn
 */
ArduinoThing.prototype['device_fqbn'] = undefined;

/**
 * The id of the device
 * @member {String} device_id
 */
ArduinoThing.prototype['device_id'] = undefined;

/**
 * The name of the attached device, if any
 * @member {String} device_name
 */
ArduinoThing.prototype['device_name'] = undefined;

/**
 * The type of the attached device, if any
 * @member {String} device_type
 */
ArduinoThing.prototype['device_type'] = undefined;

/**
 * The api reference of this thing
 * @member {String} href
 */
ArduinoThing.prototype['href'] = undefined;

/**
 * The id of the thing
 * @member {String} id
 */
ArduinoThing.prototype['id'] = undefined;

/**
 * The friendly name of the thing
 * @member {String} name
 */
ArduinoThing.prototype['name'] = undefined;

/**
 * Id of the organization the thing belongs to
 * @member {String} organization_id
 */
ArduinoThing.prototype['organization_id'] = undefined;

/**
 * ArduinoPropertyCollection is the media type for an array of ArduinoProperty (default view)
 * @member {Array.<module:model/ArduinoProperty>} properties
 */
ArduinoThing.prototype['properties'] = undefined;

/**
 * The number of properties of the thing
 * @member {Number} properties_count
 */
ArduinoThing.prototype['properties_count'] = undefined;

/**
 * The id of the attached sketch
 * @member {String} sketch_id
 */
ArduinoThing.prototype['sketch_id'] = undefined;

/**
 * Tags of the thing
 * @member {Object.<String, Object>} tags
 */
ArduinoThing.prototype['tags'] = undefined;

/**
 * Time zone of the thing
 * @member {String} timezone
 */
ArduinoThing.prototype['timezone'] = undefined;

/**
 * Update date of the thing
 * @member {Date} updated_at
 */
ArduinoThing.prototype['updated_at'] = undefined;

/**
 * The user id of the owner
 * @member {String} user_id
 */
ArduinoThing.prototype['user_id'] = undefined;

/**
 * Webhook uri
 * @member {Boolean} webhook_active
 */
ArduinoThing.prototype['webhook_active'] = undefined;

/**
 * Webhook uri
 * @member {String} webhook_uri
 */
ArduinoThing.prototype['webhook_uri'] = undefined;

/**
 * Allowed values for the <code>assistant</code> property.
 * @enum {String}
 * @readonly
 */
ArduinoThing['AssistantEnum'] = {
  /**
   * value: "ALEXA"
   * @const
   */
  "ALEXA": "ALEXA",
  /**
   * value: "GOOGLE"
   * @const
   */
  "GOOGLE": "GOOGLE",
  /**
   * value: "NONE"
   * @const
   */
  "NONE": "NONE"
};
var _default = exports["default"] = ArduinoThing;
