// @arduino/arduino-iot-client/dist/model/ArduinoDevicev2Otaupload.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(r, e) { var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (!t) { if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) { t && (r = t); var _n = 0, F = function F() {}; return { s: F, n: function n() { return _n >= r.length ? { done: !0 } : { done: !1, value: r[_n++] }; }, e: function e(r) { throw r; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var o, a = !0, u = !1; return { s: function s() { t = t.call(r); }, n: function n() { var r = t.next(); return a = r.done, r; }, e: function e(r) { u = !0, o = r; }, f: function f() { try { a || null == t["return"] || t["return"](); } finally { if (u) throw o; } } }; }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Arduino IoT Cloud API
 * Provides a set of endpoints to manage Arduino IoT Cloud **Devices**, **Things**, **Properties** and **Timeseries**. This API can be called just with any HTTP Client, or using one of these clients:  * [Javascript NPM package](https://www.npmjs.com/package/@arduino/arduino-iot-client)  * [Python PYPI Package](https://pypi.org/project/arduino-iot-client/)  * [Golang Module](https://github.com/arduino/iot-client-go)
 *
 * The version of the OpenAPI document: 2.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The ArduinoDevicev2Otaupload model module.
 * @module model/ArduinoDevicev2Otaupload
 * @version 3.0.0
 */
var ArduinoDevicev2Otaupload = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ArduinoDevicev2Otaupload</code>.
   * ArduinoDevicev2Otaupload media type (default view)
   * @alias module:model/ArduinoDevicev2Otaupload
   * @param otaVersion {Number} OTA version
   */
  function ArduinoDevicev2Otaupload(otaVersion) {
    _classCallCheck(this, ArduinoDevicev2Otaupload);
    ArduinoDevicev2Otaupload.initialize(this, otaVersion);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(ArduinoDevicev2Otaupload, null, [{
    key: "initialize",
    value: function initialize(obj, otaVersion) {
      obj['ota_version'] = otaVersion;
    }

    /**
     * Constructs a <code>ArduinoDevicev2Otaupload</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ArduinoDevicev2Otaupload} obj Optional instance to populate.
     * @return {module:model/ArduinoDevicev2Otaupload} The populated <code>ArduinoDevicev2Otaupload</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ArduinoDevicev2Otaupload();
        if (data.hasOwnProperty('file_sha')) {
          obj['file_sha'] = _ApiClient["default"].convertToType(data['file_sha'], 'String');
        }
        if (data.hasOwnProperty('ota_id')) {
          obj['ota_id'] = _ApiClient["default"].convertToType(data['ota_id'], 'String');
        }
        if (data.hasOwnProperty('ota_version')) {
          obj['ota_version'] = _ApiClient["default"].convertToType(data['ota_version'], 'Number');
        }
        if (data.hasOwnProperty('status')) {
          obj['status'] = _ApiClient["default"].convertToType(data['status'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>ArduinoDevicev2Otaupload</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ArduinoDevicev2Otaupload</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(ArduinoDevicev2Otaupload.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
        // ensure the json data is a string
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      if (data['file_sha'] && !(typeof data['file_sha'] === 'string' || data['file_sha'] instanceof String)) {
        throw new Error("Expected the field `file_sha` to be a primitive type in the JSON string but got " + data['file_sha']);
      }
      // ensure the json data is a string
      if (data['ota_id'] && !(typeof data['ota_id'] === 'string' || data['ota_id'] instanceof String)) {
        throw new Error("Expected the field `ota_id` to be a primitive type in the JSON string but got " + data['ota_id']);
      }
      // ensure the json data is a string
      if (data['status'] && !(typeof data['status'] === 'string' || data['status'] instanceof String)) {
        throw new Error("Expected the field `status` to be a primitive type in the JSON string but got " + data['status']);
      }
      return true;
    }
  }]);
}();
ArduinoDevicev2Otaupload.RequiredProperties = ["ota_version"];

/**
 * SHA256 of the uploaded file
 * @member {String} file_sha
 */
ArduinoDevicev2Otaupload.prototype['file_sha'] = undefined;

/**
 * OTA request id (only available from OTA version 2 and above)
 * @member {String} ota_id
 */
ArduinoDevicev2Otaupload.prototype['ota_id'] = undefined;

/**
 * OTA version
 * @member {Number} ota_version
 */
ArduinoDevicev2Otaupload.prototype['ota_version'] = undefined;

/**
 * OTA request status (only available from OTA version 2 and above)
 * @member {String} status
 */
ArduinoDevicev2Otaupload.prototype['status'] = undefined;
var _default = exports["default"] = ArduinoDevicev2Otaupload;
