// @arduino/arduino-iot-client/dist/model/ArduinoTemplateproperty.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(r, e) { var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (!t) { if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) { t && (r = t); var _n = 0, F = function F() {}; return { s: F, n: function n() { return _n >= r.length ? { done: !0 } : { done: !1, value: r[_n++] }; }, e: function e(r) { throw r; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var o, a = !0, u = !1; return { s: function s() { t = t.call(r); }, n: function n() { var r = t.next(); return a = r.done, r; }, e: function e(r) { u = !0, o = r; }, f: function f() { try { a || null == t["return"] || t["return"](); } finally { if (u) throw o; } } }; }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Arduino IoT Cloud API
 * Provides a set of endpoints to manage Arduino IoT Cloud **Devices**, **Things**, **Properties** and **Timeseries**. This API can be called just with any HTTP Client, or using one of these clients:  * [Javascript NPM package](https://www.npmjs.com/package/@arduino/arduino-iot-client)  * [Python PYPI Package](https://pypi.org/project/arduino-iot-client/)  * [Golang Module](https://github.com/arduino/iot-client-go)
 *
 * The version of the OpenAPI document: 2.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The ArduinoTemplateproperty model module.
 * @module model/ArduinoTemplateproperty
 * @version 3.0.0
 */
var ArduinoTemplateproperty = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ArduinoTemplateproperty</code>.
   * ArduinoTemplateproperty media type (default view)
   * @alias module:model/ArduinoTemplateproperty
   * @param name {String} The friendly name of the property
   * @param permission {String} The permission of the property
   * @param type {String} The type of the property
   * @param updateStrategy {String} The update strategy for the property value
   */
  function ArduinoTemplateproperty(name, permission, type, updateStrategy) {
    _classCallCheck(this, ArduinoTemplateproperty);
    ArduinoTemplateproperty.initialize(this, name, permission, type, updateStrategy);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(ArduinoTemplateproperty, null, [{
    key: "initialize",
    value: function initialize(obj, name, permission, type, updateStrategy) {
      obj['name'] = name;
      obj['permission'] = permission;
      obj['type'] = type;
      obj['update_strategy'] = updateStrategy;
    }

    /**
     * Constructs a <code>ArduinoTemplateproperty</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ArduinoTemplateproperty} obj Optional instance to populate.
     * @return {module:model/ArduinoTemplateproperty} The populated <code>ArduinoTemplateproperty</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ArduinoTemplateproperty();
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('permission')) {
          obj['permission'] = _ApiClient["default"].convertToType(data['permission'], 'String');
        }
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
        if (data.hasOwnProperty('update_parameter')) {
          obj['update_parameter'] = _ApiClient["default"].convertToType(data['update_parameter'], 'Number');
        }
        if (data.hasOwnProperty('update_strategy')) {
          obj['update_strategy'] = _ApiClient["default"].convertToType(data['update_strategy'], 'String');
        }
        if (data.hasOwnProperty('variable_name')) {
          obj['variable_name'] = _ApiClient["default"].convertToType(data['variable_name'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>ArduinoTemplateproperty</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ArduinoTemplateproperty</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(ArduinoTemplateproperty.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
        // ensure the json data is a string
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      if (data['id'] && !(typeof data['id'] === 'string' || data['id'] instanceof String)) {
        throw new Error("Expected the field `id` to be a primitive type in the JSON string but got " + data['id']);
      }
      // ensure the json data is a string
      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      }
      // ensure the json data is a string
      if (data['permission'] && !(typeof data['permission'] === 'string' || data['permission'] instanceof String)) {
        throw new Error("Expected the field `permission` to be a primitive type in the JSON string but got " + data['permission']);
      }
      // ensure the json data is a string
      if (data['type'] && !(typeof data['type'] === 'string' || data['type'] instanceof String)) {
        throw new Error("Expected the field `type` to be a primitive type in the JSON string but got " + data['type']);
      }
      // ensure the json data is a string
      if (data['update_strategy'] && !(typeof data['update_strategy'] === 'string' || data['update_strategy'] instanceof String)) {
        throw new Error("Expected the field `update_strategy` to be a primitive type in the JSON string but got " + data['update_strategy']);
      }
      // ensure the json data is a string
      if (data['variable_name'] && !(typeof data['variable_name'] === 'string' || data['variable_name'] instanceof String)) {
        throw new Error("Expected the field `variable_name` to be a primitive type in the JSON string but got " + data['variable_name']);
      }
      return true;
    }
  }]);
}();
ArduinoTemplateproperty.RequiredProperties = ["name", "permission", "type", "update_strategy"];

/**
 * The friendly id of the property
 * @member {String} id
 */
ArduinoTemplateproperty.prototype['id'] = undefined;

/**
 * The friendly name of the property
 * @member {String} name
 */
ArduinoTemplateproperty.prototype['name'] = undefined;

/**
 * The permission of the property
 * @member {String} permission
 */
ArduinoTemplateproperty.prototype['permission'] = undefined;

/**
 * The type of the property
 * @member {String} type
 */
ArduinoTemplateproperty.prototype['type'] = undefined;

/**
 * The update frequency in seconds, or the amount of the property has to change in order to trigger an update
 * @member {Number} update_parameter
 */
ArduinoTemplateproperty.prototype['update_parameter'] = undefined;

/**
 * The update strategy for the property value
 * @member {String} update_strategy
 */
ArduinoTemplateproperty.prototype['update_strategy'] = undefined;

/**
 * The sketch variable name of the property
 * @member {String} variable_name
 */
ArduinoTemplateproperty.prototype['variable_name'] = undefined;
var _default = exports["default"] = ArduinoTemplateproperty;
