// @arduino/arduino-iot-client/dist/model/ArduinoDevicev2propertyvalueValueStatistics.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Arduino IoT Cloud API
 * Provides a set of endpoints to manage Arduino IoT Cloud **Devices**, **Things**, **Properties** and **Timeseries**. This API can be called just with any HTTP Client, or using one of these clients:  * [Javascript NPM package](https://www.npmjs.com/package/@arduino/arduino-iot-client)  * [Python PYPI Package](https://pypi.org/project/arduino-iot-client/)  * [Golang Module](https://github.com/arduino/iot-client-go)
 *
 * The version of the OpenAPI document: 2.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The ArduinoDevicev2propertyvalueValueStatistics model module.
 * @module model/ArduinoDevicev2propertyvalueValueStatistics
 * @version 3.0.0
 */
var ArduinoDevicev2propertyvalueValueStatistics = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ArduinoDevicev2propertyvalueValueStatistics</code>.
   * @alias module:model/ArduinoDevicev2propertyvalueValueStatistics
   */
  function ArduinoDevicev2propertyvalueValueStatistics() {
    _classCallCheck(this, ArduinoDevicev2propertyvalueValueStatistics);
    ArduinoDevicev2propertyvalueValueStatistics.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(ArduinoDevicev2propertyvalueValueStatistics, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>ArduinoDevicev2propertyvalueValueStatistics</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ArduinoDevicev2propertyvalueValueStatistics} obj Optional instance to populate.
     * @return {module:model/ArduinoDevicev2propertyvalueValueStatistics} The populated <code>ArduinoDevicev2propertyvalueValueStatistics</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ArduinoDevicev2propertyvalueValueStatistics();
        if (data.hasOwnProperty('adr')) {
          obj['adr'] = _ApiClient["default"].convertToType(data['adr'], 'Number');
        }
        if (data.hasOwnProperty('channel')) {
          obj['channel'] = _ApiClient["default"].convertToType(data['channel'], 'Number');
        }
        if (data.hasOwnProperty('duplicate')) {
          obj['duplicate'] = _ApiClient["default"].convertToType(data['duplicate'], 'Number');
        }
        if (data.hasOwnProperty('freq')) {
          obj['freq'] = _ApiClient["default"].convertToType(data['freq'], 'Number');
        }
        if (data.hasOwnProperty('modBW')) {
          obj['modBW'] = _ApiClient["default"].convertToType(data['modBW'], 'Number');
        }
        if (data.hasOwnProperty('rssi')) {
          obj['rssi'] = _ApiClient["default"].convertToType(data['rssi'], 'Number');
        }
        if (data.hasOwnProperty('seqno')) {
          obj['seqno'] = _ApiClient["default"].convertToType(data['seqno'], 'Number');
        }
        if (data.hasOwnProperty('sf')) {
          obj['sf'] = _ApiClient["default"].convertToType(data['sf'], 'Number');
        }
        if (data.hasOwnProperty('snr')) {
          obj['snr'] = _ApiClient["default"].convertToType(data['snr'], 'Number');
        }
        if (data.hasOwnProperty('time')) {
          obj['time'] = _ApiClient["default"].convertToType(data['time'], 'Number');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>ArduinoDevicev2propertyvalueValueStatistics</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ArduinoDevicev2propertyvalueValueStatistics</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      return true;
    }
  }]);
}();
/**
 * @member {Number} adr
 */
ArduinoDevicev2propertyvalueValueStatistics.prototype['adr'] = undefined;

/**
 * @member {Number} channel
 */
ArduinoDevicev2propertyvalueValueStatistics.prototype['channel'] = undefined;

/**
 * @member {Number} duplicate
 */
ArduinoDevicev2propertyvalueValueStatistics.prototype['duplicate'] = undefined;

/**
 * @member {Number} freq
 */
ArduinoDevicev2propertyvalueValueStatistics.prototype['freq'] = undefined;

/**
 * @member {Number} modBW
 */
ArduinoDevicev2propertyvalueValueStatistics.prototype['modBW'] = undefined;

/**
 * @member {Number} rssi
 */
ArduinoDevicev2propertyvalueValueStatistics.prototype['rssi'] = undefined;

/**
 * @member {Number} seqno
 */
ArduinoDevicev2propertyvalueValueStatistics.prototype['seqno'] = undefined;

/**
 * @member {Number} sf
 */
ArduinoDevicev2propertyvalueValueStatistics.prototype['sf'] = undefined;

/**
 * @member {Number} snr
 */
ArduinoDevicev2propertyvalueValueStatistics.prototype['snr'] = undefined;

/**
 * @member {Number} time
 */
ArduinoDevicev2propertyvalueValueStatistics.prototype['time'] = undefined;
var _default = exports["default"] = ArduinoDevicev2propertyvalueValueStatistics;
