// @arduino/arduino-iot-client/dist/model/ArduinoThingtemplate.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ArduinoDevicev2templatedevice = _interopRequireDefault(require("./ArduinoDevicev2templatedevice"));
var _ArduinoTemplateproperty = _interopRequireDefault(require("./ArduinoTemplateproperty"));
var _Tag = _interopRequireDefault(require("./Tag"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(r, e) { var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (!t) { if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) { t && (r = t); var _n = 0, F = function F() {}; return { s: F, n: function n() { return _n >= r.length ? { done: !0 } : { done: !1, value: r[_n++] }; }, e: function e(r) { throw r; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var o, a = !0, u = !1; return { s: function s() { t = t.call(r); }, n: function n() { var r = t.next(); return a = r.done, r; }, e: function e(r) { u = !0, o = r; }, f: function f() { try { a || null == t["return"] || t["return"](); } finally { if (u) throw o; } } }; }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Arduino IoT Cloud API
 * Provides a set of endpoints to manage Arduino IoT Cloud **Devices**, **Things**, **Properties** and **Timeseries**. This API can be called just with any HTTP Client, or using one of these clients:  * [Javascript NPM package](https://www.npmjs.com/package/@arduino/arduino-iot-client)  * [Python PYPI Package](https://pypi.org/project/arduino-iot-client/)  * [Golang Module](https://github.com/arduino/iot-client-go)
 *
 * The version of the OpenAPI document: 2.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The ArduinoThingtemplate model module.
 * @module model/ArduinoThingtemplate
 * @version 3.0.0
 */
var ArduinoThingtemplate = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ArduinoThingtemplate</code>.
   * ArduinoThingtemplate media type (default view)
   * @alias module:model/ArduinoThingtemplate
   * @param name {String} The friendly name of the thing
   * @param timezone {String} Time zone of the thing
   */
  function ArduinoThingtemplate(name, timezone) {
    _classCallCheck(this, ArduinoThingtemplate);
    ArduinoThingtemplate.initialize(this, name, timezone);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(ArduinoThingtemplate, null, [{
    key: "initialize",
    value: function initialize(obj, name, timezone) {
      obj['name'] = name;
      obj['timezone'] = timezone;
    }

    /**
     * Constructs a <code>ArduinoThingtemplate</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ArduinoThingtemplate} obj Optional instance to populate.
     * @return {module:model/ArduinoThingtemplate} The populated <code>ArduinoThingtemplate</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ArduinoThingtemplate();
        if (data.hasOwnProperty('device_metadata')) {
          obj['device_metadata'] = _ArduinoDevicev2templatedevice["default"].constructFromObject(data['device_metadata']);
        }
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('organization_id')) {
          obj['organization_id'] = _ApiClient["default"].convertToType(data['organization_id'], 'String');
        }
        if (data.hasOwnProperty('sketch_template')) {
          obj['sketch_template'] = _ApiClient["default"].convertToType(data['sketch_template'], 'String');
        }
        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], [_Tag["default"]]);
        }
        if (data.hasOwnProperty('timezone')) {
          obj['timezone'] = _ApiClient["default"].convertToType(data['timezone'], 'String');
        }
        if (data.hasOwnProperty('variables')) {
          obj['variables'] = _ApiClient["default"].convertToType(data['variables'], [_ArduinoTemplateproperty["default"]]);
        }
        if (data.hasOwnProperty('webhook_uri')) {
          obj['webhook_uri'] = _ApiClient["default"].convertToType(data['webhook_uri'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>ArduinoThingtemplate</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ArduinoThingtemplate</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(ArduinoThingtemplate.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
        // validate the optional field `device_metadata`
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      if (data['device_metadata']) {
        // data not null
        _ArduinoDevicev2templatedevice["default"].validateJSON(data['device_metadata']);
      }
      // ensure the json data is a string
      if (data['id'] && !(typeof data['id'] === 'string' || data['id'] instanceof String)) {
        throw new Error("Expected the field `id` to be a primitive type in the JSON string but got " + data['id']);
      }
      // ensure the json data is a string
      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      }
      // ensure the json data is a string
      if (data['organization_id'] && !(typeof data['organization_id'] === 'string' || data['organization_id'] instanceof String)) {
        throw new Error("Expected the field `organization_id` to be a primitive type in the JSON string but got " + data['organization_id']);
      }
      // ensure the json data is a string
      if (data['sketch_template'] && !(typeof data['sketch_template'] === 'string' || data['sketch_template'] instanceof String)) {
        throw new Error("Expected the field `sketch_template` to be a primitive type in the JSON string but got " + data['sketch_template']);
      }
      if (data['tags']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['tags'])) {
          throw new Error("Expected the field `tags` to be an array in the JSON data but got " + data['tags']);
        }
        // validate the optional field `tags` (array)
        var _iterator2 = _createForOfIteratorHelper(data['tags']),
          _step2;
        try {
          for (_iterator2.s(); !(_step2 = _iterator2.n()).done;) {
            var item = _step2.value;
            _Tag["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator2.e(err);
        } finally {
          _iterator2.f();
        }
        ;
      }
      // ensure the json data is a string
      if (data['timezone'] && !(typeof data['timezone'] === 'string' || data['timezone'] instanceof String)) {
        throw new Error("Expected the field `timezone` to be a primitive type in the JSON string but got " + data['timezone']);
      }
      if (data['variables']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['variables'])) {
          throw new Error("Expected the field `variables` to be an array in the JSON data but got " + data['variables']);
        }
        // validate the optional field `variables` (array)
        var _iterator3 = _createForOfIteratorHelper(data['variables']),
          _step3;
        try {
          for (_iterator3.s(); !(_step3 = _iterator3.n()).done;) {
            var _item = _step3.value;
            _ArduinoTemplateproperty["default"].validateJSON(_item);
          }
        } catch (err) {
          _iterator3.e(err);
        } finally {
          _iterator3.f();
        }
        ;
      }
      // ensure the json data is a string
      if (data['webhook_uri'] && !(typeof data['webhook_uri'] === 'string' || data['webhook_uri'] instanceof String)) {
        throw new Error("Expected the field `webhook_uri` to be a primitive type in the JSON string but got " + data['webhook_uri']);
      }
      return true;
    }
  }]);
}();
ArduinoThingtemplate.RequiredProperties = ["name", "timezone"];

/**
 * @member {module:model/ArduinoDevicev2templatedevice} device_metadata
 */
ArduinoThingtemplate.prototype['device_metadata'] = undefined;

/**
 * The friendly id of the thing
 * @member {String} id
 */
ArduinoThingtemplate.prototype['id'] = undefined;

/**
 * The friendly name of the thing
 * @member {String} name
 */
ArduinoThingtemplate.prototype['name'] = undefined;

/**
 * Id of the organization the thing belongs to
 * @member {String} organization_id
 */
ArduinoThingtemplate.prototype['organization_id'] = undefined;

/**
 * The ID of the template's sketch
 * @member {String} sketch_template
 */
ArduinoThingtemplate.prototype['sketch_template'] = undefined;

/**
 * Tags of the thing
 * @member {Array.<module:model/Tag>} tags
 */
ArduinoThingtemplate.prototype['tags'] = undefined;

/**
 * Time zone of the thing
 * @member {String} timezone
 */
ArduinoThingtemplate.prototype['timezone'] = undefined;

/**
 * ArduinoTemplatepropertyCollection is the media type for an array of ArduinoTemplateproperty (default view)
 * @member {Array.<module:model/ArduinoTemplateproperty>} variables
 */
ArduinoThingtemplate.prototype['variables'] = undefined;

/**
 * Webhook uri
 * @member {String} webhook_uri
 */
ArduinoThingtemplate.prototype['webhook_uri'] = undefined;
var _default = exports["default"] = ArduinoThingtemplate;
