// @arduino/arduino-iot-client/dist/model/ArduinoWidgetv2.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ArduinoLinkedvariable = _interopRequireDefault(require("./ArduinoLinkedvariable"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(r, e) { var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (!t) { if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) { t && (r = t); var _n = 0, F = function F() {}; return { s: F, n: function n() { return _n >= r.length ? { done: !0 } : { done: !1, value: r[_n++] }; }, e: function e(r) { throw r; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var o, a = !0, u = !1; return { s: function s() { t = t.call(r); }, n: function n() { var r = t.next(); return a = r.done, r; }, e: function e(r) { u = !0, o = r; }, f: function f() { try { a || null == t["return"] || t["return"](); } finally { if (u) throw o; } } }; }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Arduino IoT Cloud API
 * Provides a set of endpoints to manage Arduino IoT Cloud **Devices**, **Things**, **Properties** and **Timeseries**. This API can be called just with any HTTP Client, or using one of these clients:  * [Javascript NPM package](https://www.npmjs.com/package/@arduino/arduino-iot-client)  * [Python PYPI Package](https://pypi.org/project/arduino-iot-client/)  * [Golang Module](https://github.com/arduino/iot-client-go)
 *
 * The version of the OpenAPI document: 2.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The ArduinoWidgetv2 model module.
 * @module model/ArduinoWidgetv2
 * @version 3.0.0
 */
var ArduinoWidgetv2 = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ArduinoWidgetv2</code>.
   * ArduinoWidgetv2 media type (default view)
   * @alias module:model/ArduinoWidgetv2
   * @param height {Number} Widget current height for desktop
   * @param id {String} The UUID of the widget, set by client
   * @param options {Object.<String, Object>} Widget options
   * @param type {String} The type of the widget
   * @param width {Number} Widget current width for desktop
   * @param x {Number} Widget x position for desktop
   * @param y {Number} Widget y position for desktop
   */
  function ArduinoWidgetv2(height, id, options, type, width, x, y) {
    _classCallCheck(this, ArduinoWidgetv2);
    ArduinoWidgetv2.initialize(this, height, id, options, type, width, x, y);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(ArduinoWidgetv2, null, [{
    key: "initialize",
    value: function initialize(obj, height, id, options, type, width, x, y) {
      obj['height'] = height;
      obj['id'] = id;
      obj['options'] = options;
      obj['type'] = type;
      obj['width'] = width;
      obj['x'] = x;
      obj['y'] = y;
    }

    /**
     * Constructs a <code>ArduinoWidgetv2</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ArduinoWidgetv2} obj Optional instance to populate.
     * @return {module:model/ArduinoWidgetv2} The populated <code>ArduinoWidgetv2</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ArduinoWidgetv2();
        if (data.hasOwnProperty('has_permission_incompatibility')) {
          obj['has_permission_incompatibility'] = _ApiClient["default"].convertToType(data['has_permission_incompatibility'], 'Boolean');
        }
        if (data.hasOwnProperty('has_type_incompatibility')) {
          obj['has_type_incompatibility'] = _ApiClient["default"].convertToType(data['has_type_incompatibility'], 'Boolean');
        }
        if (data.hasOwnProperty('has_unlinked_variable')) {
          obj['has_unlinked_variable'] = _ApiClient["default"].convertToType(data['has_unlinked_variable'], 'Boolean');
        }
        if (data.hasOwnProperty('height')) {
          obj['height'] = _ApiClient["default"].convertToType(data['height'], 'Number');
        }
        if (data.hasOwnProperty('height_mobile')) {
          obj['height_mobile'] = _ApiClient["default"].convertToType(data['height_mobile'], 'Number');
        }
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('options')) {
          obj['options'] = _ApiClient["default"].convertToType(data['options'], {
            'String': Object
          });
        }
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
        if (data.hasOwnProperty('variables')) {
          obj['variables'] = _ApiClient["default"].convertToType(data['variables'], [_ArduinoLinkedvariable["default"]]);
        }
        if (data.hasOwnProperty('width')) {
          obj['width'] = _ApiClient["default"].convertToType(data['width'], 'Number');
        }
        if (data.hasOwnProperty('width_mobile')) {
          obj['width_mobile'] = _ApiClient["default"].convertToType(data['width_mobile'], 'Number');
        }
        if (data.hasOwnProperty('x')) {
          obj['x'] = _ApiClient["default"].convertToType(data['x'], 'Number');
        }
        if (data.hasOwnProperty('x_mobile')) {
          obj['x_mobile'] = _ApiClient["default"].convertToType(data['x_mobile'], 'Number');
        }
        if (data.hasOwnProperty('y')) {
          obj['y'] = _ApiClient["default"].convertToType(data['y'], 'Number');
        }
        if (data.hasOwnProperty('y_mobile')) {
          obj['y_mobile'] = _ApiClient["default"].convertToType(data['y_mobile'], 'Number');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>ArduinoWidgetv2</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ArduinoWidgetv2</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(ArduinoWidgetv2.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
        // ensure the json data is a string
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      if (data['id'] && !(typeof data['id'] === 'string' || data['id'] instanceof String)) {
        throw new Error("Expected the field `id` to be a primitive type in the JSON string but got " + data['id']);
      }
      // ensure the json data is a string
      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      }
      // ensure the json data is a string
      if (data['type'] && !(typeof data['type'] === 'string' || data['type'] instanceof String)) {
        throw new Error("Expected the field `type` to be a primitive type in the JSON string but got " + data['type']);
      }
      if (data['variables']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['variables'])) {
          throw new Error("Expected the field `variables` to be an array in the JSON data but got " + data['variables']);
        }
        // validate the optional field `variables` (array)
        var _iterator2 = _createForOfIteratorHelper(data['variables']),
          _step2;
        try {
          for (_iterator2.s(); !(_step2 = _iterator2.n()).done;) {
            var item = _step2.value;
            _ArduinoLinkedvariable["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator2.e(err);
        } finally {
          _iterator2.f();
        }
        ;
      }
      return true;
    }
  }]);
}();
ArduinoWidgetv2.RequiredProperties = ["height", "id", "options", "type", "width", "x", "y"];

/**
 * True if the linked variables permissions are incompatible with the widget
 * @member {Boolean} has_permission_incompatibility
 */
ArduinoWidgetv2.prototype['has_permission_incompatibility'] = undefined;

/**
 * True if the linked variables types are incompatible with the widget
 * @member {Boolean} has_type_incompatibility
 */
ArduinoWidgetv2.prototype['has_type_incompatibility'] = undefined;

/**
 * If it's true the widget is linked to a soft-deleted variable
 * @member {Boolean} has_unlinked_variable
 */
ArduinoWidgetv2.prototype['has_unlinked_variable'] = undefined;

/**
 * Widget current height for desktop
 * @member {Number} height
 */
ArduinoWidgetv2.prototype['height'] = undefined;

/**
 * Widget current height for mobile
 * @member {Number} height_mobile
 */
ArduinoWidgetv2.prototype['height_mobile'] = undefined;

/**
 * The UUID of the widget, set by client
 * @member {String} id
 */
ArduinoWidgetv2.prototype['id'] = undefined;

/**
 * The name of the widget
 * @member {String} name
 */
ArduinoWidgetv2.prototype['name'] = undefined;

/**
 * Widget options
 * @member {Object.<String, Object>} options
 */
ArduinoWidgetv2.prototype['options'] = undefined;

/**
 * The type of the widget
 * @member {String} type
 */
ArduinoWidgetv2.prototype['type'] = undefined;

/**
 * ArduinoLinkedvariableCollection is the media type for an array of ArduinoLinkedvariable (default view)
 * @member {Array.<module:model/ArduinoLinkedvariable>} variables
 */
ArduinoWidgetv2.prototype['variables'] = undefined;

/**
 * Widget current width for desktop
 * @member {Number} width
 */
ArduinoWidgetv2.prototype['width'] = undefined;

/**
 * Widget current width for mobile
 * @member {Number} width_mobile
 */
ArduinoWidgetv2.prototype['width_mobile'] = undefined;

/**
 * Widget x position for desktop
 * @member {Number} x
 */
ArduinoWidgetv2.prototype['x'] = undefined;

/**
 * Widget x position for mobile
 * @member {Number} x_mobile
 */
ArduinoWidgetv2.prototype['x_mobile'] = undefined;

/**
 * Widget y position for desktop
 * @member {Number} y
 */
ArduinoWidgetv2.prototype['y'] = undefined;

/**
 * Widget y position for mobile
 * @member {Number} y_mobile
 */
ArduinoWidgetv2.prototype['y_mobile'] = undefined;
var _default = exports["default"] = ArduinoWidgetv2;
