// @arduino/arduino-iot-client/dist/model/ArduinoDevicev2propertyvalueValue.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ArduinoDevicev2propertyvalueValueStatistics = _interopRequireDefault(require("./ArduinoDevicev2propertyvalueValueStatistics"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Arduino IoT Cloud API
 * Provides a set of endpoints to manage Arduino IoT Cloud **Devices**, **Things**, **Properties** and **Timeseries**. This API can be called just with any HTTP Client, or using one of these clients:  * [Javascript NPM package](https://www.npmjs.com/package/@arduino/arduino-iot-client)  * [Python PYPI Package](https://pypi.org/project/arduino-iot-client/)  * [Golang Module](https://github.com/arduino/iot-client-go)
 *
 * The version of the OpenAPI document: 2.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The ArduinoDevicev2propertyvalueValue model module.
 * @module model/ArduinoDevicev2propertyvalueValue
 * @version 3.0.0
 */
var ArduinoDevicev2propertyvalueValue = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ArduinoDevicev2propertyvalueValue</code>.
   * @alias module:model/ArduinoDevicev2propertyvalueValue
   */
  function ArduinoDevicev2propertyvalueValue() {
    _classCallCheck(this, ArduinoDevicev2propertyvalueValue);
    ArduinoDevicev2propertyvalueValue.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(ArduinoDevicev2propertyvalueValue, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>ArduinoDevicev2propertyvalueValue</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ArduinoDevicev2propertyvalueValue} obj Optional instance to populate.
     * @return {module:model/ArduinoDevicev2propertyvalueValue} The populated <code>ArduinoDevicev2propertyvalueValue</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ArduinoDevicev2propertyvalueValue();
        if (data.hasOwnProperty('payload')) {
          obj['payload'] = _ApiClient["default"].convertToType(data['payload'], 'String');
        }
        if (data.hasOwnProperty('seqno')) {
          obj['seqno'] = _ApiClient["default"].convertToType(data['seqno'], 'Number');
        }
        if (data.hasOwnProperty('statistics')) {
          obj['statistics'] = _ArduinoDevicev2propertyvalueValueStatistics["default"].constructFromObject(data['statistics']);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>ArduinoDevicev2propertyvalueValue</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ArduinoDevicev2propertyvalueValue</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['payload'] && !(typeof data['payload'] === 'string' || data['payload'] instanceof String)) {
        throw new Error("Expected the field `payload` to be a primitive type in the JSON string but got " + data['payload']);
      }
      // validate the optional field `statistics`
      if (data['statistics']) {
        // data not null
        _ArduinoDevicev2propertyvalueValueStatistics["default"].validateJSON(data['statistics']);
      }
      return true;
    }
  }]);
}();
/**
 * @member {String} payload
 */
ArduinoDevicev2propertyvalueValue.prototype['payload'] = undefined;

/**
 * @member {Number} seqno
 */
ArduinoDevicev2propertyvalueValue.prototype['seqno'] = undefined;

/**
 * @member {module:model/ArduinoDevicev2propertyvalueValueStatistics} statistics
 */
ArduinoDevicev2propertyvalueValue.prototype['statistics'] = undefined;
var _default = exports["default"] = ArduinoDevicev2propertyvalueValue;
