// @arduino/arduino-iot-client/dist/model/BatchQueryRequestMediaV1.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(r, e) { var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (!t) { if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) { t && (r = t); var _n = 0, F = function F() {}; return { s: F, n: function n() { return _n >= r.length ? { done: !0 } : { done: !1, value: r[_n++] }; }, e: function e(r) { throw r; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var o, a = !0, u = !1; return { s: function s() { t = t.call(r); }, n: function n() { var r = t.next(); return a = r.done, r; }, e: function e(r) { u = !0, o = r; }, f: function f() { try { a || null == t["return"] || t["return"](); } finally { if (u) throw o; } } }; }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Arduino IoT Cloud API
 * Provides a set of endpoints to manage Arduino IoT Cloud **Devices**, **Things**, **Properties** and **Timeseries**. This API can be called just with any HTTP Client, or using one of these clients:  * [Javascript NPM package](https://www.npmjs.com/package/@arduino/arduino-iot-client)  * [Python PYPI Package](https://pypi.org/project/arduino-iot-client/)  * [Golang Module](https://github.com/arduino/iot-client-go)
 *
 * The version of the OpenAPI document: 2.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The BatchQueryRequestMediaV1 model module.
 * @module model/BatchQueryRequestMediaV1
 * @version 3.0.0
 */
var BatchQueryRequestMediaV1 = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>BatchQueryRequestMediaV1</code>.
   * @alias module:model/BatchQueryRequestMediaV1
   * @param from {Date} From timestamp
   * @param q {String} Data selection query (e.g. property.2a99729d-2556-4220-a139-023348a1e6b5 or thing.95717675-4786-4ffc-afcc-799777755391)
   * @param to {Date} To timestamp
   */
  function BatchQueryRequestMediaV1(from, q, to) {
    _classCallCheck(this, BatchQueryRequestMediaV1);
    BatchQueryRequestMediaV1.initialize(this, from, q, to);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(BatchQueryRequestMediaV1, null, [{
    key: "initialize",
    value: function initialize(obj, from, q, to) {
      obj['from'] = from;
      obj['q'] = q;
      obj['to'] = to;
    }

    /**
     * Constructs a <code>BatchQueryRequestMediaV1</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/BatchQueryRequestMediaV1} obj Optional instance to populate.
     * @return {module:model/BatchQueryRequestMediaV1} The populated <code>BatchQueryRequestMediaV1</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new BatchQueryRequestMediaV1();
        if (data.hasOwnProperty('aggregation')) {
          obj['aggregation'] = _ApiClient["default"].convertToType(data['aggregation'], 'String');
        }
        if (data.hasOwnProperty('from')) {
          obj['from'] = _ApiClient["default"].convertToType(data['from'], 'Date');
        }
        if (data.hasOwnProperty('interval')) {
          obj['interval'] = _ApiClient["default"].convertToType(data['interval'], 'Number');
        }
        if (data.hasOwnProperty('q')) {
          obj['q'] = _ApiClient["default"].convertToType(data['q'], 'String');
        }
        if (data.hasOwnProperty('series_limit')) {
          obj['series_limit'] = _ApiClient["default"].convertToType(data['series_limit'], 'Number');
        }
        if (data.hasOwnProperty('to')) {
          obj['to'] = _ApiClient["default"].convertToType(data['to'], 'Date');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>BatchQueryRequestMediaV1</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>BatchQueryRequestMediaV1</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(BatchQueryRequestMediaV1.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
        // ensure the json data is a string
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      if (data['aggregation'] && !(typeof data['aggregation'] === 'string' || data['aggregation'] instanceof String)) {
        throw new Error("Expected the field `aggregation` to be a primitive type in the JSON string but got " + data['aggregation']);
      }
      // ensure the json data is a string
      if (data['q'] && !(typeof data['q'] === 'string' || data['q'] instanceof String)) {
        throw new Error("Expected the field `q` to be a primitive type in the JSON string but got " + data['q']);
      }
      return true;
    }
  }]);
}();
BatchQueryRequestMediaV1.RequiredProperties = ["from", "q", "to"];

/**
 * Aggregation statistic function. For numeric values, AVG statistic is used by default. PCT_X compute the Xth approximate percentile (e.g. PCT_95 is the 95th approximate percentile). For boolean, BOOL_OR statistic is used as default.
 * @member {module:model/BatchQueryRequestMediaV1.AggregationEnum} aggregation
 */
BatchQueryRequestMediaV1.prototype['aggregation'] = undefined;

/**
 * From timestamp
 * @member {Date} from
 */
BatchQueryRequestMediaV1.prototype['from'] = undefined;

/**
 * Resolution in seconds (max allowed: 86400)
 * @member {Number} interval
 */
BatchQueryRequestMediaV1.prototype['interval'] = undefined;

/**
 * Data selection query (e.g. property.2a99729d-2556-4220-a139-023348a1e6b5 or thing.95717675-4786-4ffc-afcc-799777755391)
 * @member {String} q
 */
BatchQueryRequestMediaV1.prototype['q'] = undefined;

/**
 * Maximum number of values returned after data aggregation, if any (default: 300, limit: 1000 - 10000 in case of thing query)
 * @member {Number} series_limit
 */
BatchQueryRequestMediaV1.prototype['series_limit'] = undefined;

/**
 * To timestamp
 * @member {Date} to
 */
BatchQueryRequestMediaV1.prototype['to'] = undefined;

/**
 * Allowed values for the <code>aggregation</code> property.
 * @enum {String}
 * @readonly
 */
BatchQueryRequestMediaV1['AggregationEnum'] = {
  /**
   * value: "AVG"
   * @const
   */
  "AVG": "AVG",
  /**
   * value: "MIN"
   * @const
   */
  "MIN": "MIN",
  /**
   * value: "MAX"
   * @const
   */
  "MAX": "MAX",
  /**
   * value: "SUM"
   * @const
   */
  "SUM": "SUM",
  /**
   * value: "COUNT"
   * @const
   */
  "COUNT": "COUNT",
  /**
   * value: "PCT_99"
   * @const
   */
  "PCT_99": "PCT_99",
  /**
   * value: "PCT_95"
   * @const
   */
  "PCT_95": "PCT_95",
  /**
   * value: "PCT_90"
   * @const
   */
  "PCT_90": "PCT_90",
  /**
   * value: "PCT_75"
   * @const
   */
  "PCT_75": "PCT_75",
  /**
   * value: "PCT_50"
   * @const
   */
  "PCT_50": "PCT_50",
  /**
   * value: "PCT_15"
   * @const
   */
  "PCT_15": "PCT_15",
  /**
   * value: "PCT_5"
   * @const
   */
  "PCT_5": "PCT_5",
  /**
   * value: "BOOL_OR"
   * @const
   */
  "BOOL_OR": "BOOL_OR",
  /**
   * value: "BOOL_AND"
   * @const
   */
  "BOOL_AND": "BOOL_AND"
};
var _default = exports["default"] = BatchQueryRequestMediaV1;
