// @arduino/arduino-iot-client/dist/model/BatchQueryResponseSeriesMediaV1.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Iot API
 * Collection of all public API endpoints.
 *
 * The version of the OpenAPI document: 2.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The BatchQueryResponseSeriesMediaV1 model module.
 * @module model/BatchQueryResponseSeriesMediaV1
 * @version 0.0.1
 */
var BatchQueryResponseSeriesMediaV1 = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>BatchQueryResponseSeriesMediaV1</code>.
   * @alias module:model/BatchQueryResponseSeriesMediaV1
   * @param aggr {String} Aggregation type
   * @param expression {String} Query
   * @param metric {String} Metric name
   */
  function BatchQueryResponseSeriesMediaV1(aggr, expression, metric) {
    _classCallCheck(this, BatchQueryResponseSeriesMediaV1);
    BatchQueryResponseSeriesMediaV1.initialize(this, aggr, expression, metric);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(BatchQueryResponseSeriesMediaV1, null, [{
    key: "initialize",
    value: function initialize(obj, aggr, expression, metric) {
      obj['aggr'] = aggr;
      obj['expression'] = expression;
      obj['metric'] = metric;
    }

    /**
     * Constructs a <code>BatchQueryResponseSeriesMediaV1</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/BatchQueryResponseSeriesMediaV1} obj Optional instance to populate.
     * @return {module:model/BatchQueryResponseSeriesMediaV1} The populated <code>BatchQueryResponseSeriesMediaV1</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new BatchQueryResponseSeriesMediaV1();
        if (data.hasOwnProperty('aggr')) {
          obj['aggr'] = _ApiClient["default"].convertToType(data['aggr'], 'String');
        }
        if (data.hasOwnProperty('expression')) {
          obj['expression'] = _ApiClient["default"].convertToType(data['expression'], 'String');
        }
        if (data.hasOwnProperty('metric')) {
          obj['metric'] = _ApiClient["default"].convertToType(data['metric'], 'String');
        }
      }
      return obj;
    }
  }]);
}();
/**
 * Aggregation type
 * @member {String} aggr
 */
BatchQueryResponseSeriesMediaV1.prototype['aggr'] = undefined;

/**
 * Query
 * @member {String} expression
 */
BatchQueryResponseSeriesMediaV1.prototype['expression'] = undefined;

/**
 * Metric name
 * @member {String} metric
 */
BatchQueryResponseSeriesMediaV1.prototype['metric'] = undefined;
var _default = exports["default"] = BatchQueryResponseSeriesMediaV1;
