// @arduino/arduino-iot-client/dist/model/CreateAction.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _EmailAction = _interopRequireDefault(require("./EmailAction"));
var _PushAction = _interopRequireDefault(require("./PushAction"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(r, e) { var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (!t) { if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) { t && (r = t); var _n = 0, F = function F() {}; return { s: F, n: function n() { return _n >= r.length ? { done: !0 } : { done: !1, value: r[_n++] }; }, e: function e(r) { throw r; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var o, a = !0, u = !1; return { s: function s() { t = t.call(r); }, n: function n() { var r = t.next(); return a = r.done, r; }, e: function e(r) { u = !0, o = r; }, f: function f() { try { a || null == t["return"] || t["return"](); } finally { if (u) throw o; } } }; }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Arduino IoT Cloud API
 * Provides a set of endpoints to manage Arduino IoT Cloud **Devices**, **Things**, **Properties** and **Timeseries**. This API can be called just with any HTTP Client, or using one of these clients:  * [Javascript NPM package](https://www.npmjs.com/package/@arduino/arduino-iot-client)  * [Python PYPI Package](https://pypi.org/project/arduino-iot-client/)  * [Golang Module](https://github.com/arduino/iot-client-go)
 *
 * The version of the OpenAPI document: 2.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The CreateAction model module.
 * @module model/CreateAction
 * @version 3.0.0
 */
var CreateAction = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateAction</code>.
   * @alias module:model/CreateAction
   * @param kind {module:model/CreateAction.KindEnum} The kind of the action
   * @param name {String} The name of the action
   */
  function CreateAction(kind, name) {
    _classCallCheck(this, CreateAction);
    CreateAction.initialize(this, kind, name);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(CreateAction, null, [{
    key: "initialize",
    value: function initialize(obj, kind, name) {
      obj['kind'] = kind;
      obj['name'] = name;
    }

    /**
     * Constructs a <code>CreateAction</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateAction} obj Optional instance to populate.
     * @return {module:model/CreateAction} The populated <code>CreateAction</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateAction();
        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }
        if (data.hasOwnProperty('email')) {
          obj['email'] = _EmailAction["default"].constructFromObject(data['email']);
        }
        if (data.hasOwnProperty('kind')) {
          obj['kind'] = _ApiClient["default"].convertToType(data['kind'], 'String');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('organization_id')) {
          obj['organization_id'] = _ApiClient["default"].convertToType(data['organization_id'], 'String');
        }
        if (data.hasOwnProperty('push_notification')) {
          obj['push_notification'] = _PushAction["default"].constructFromObject(data['push_notification']);
        }
        if (data.hasOwnProperty('trigger_id')) {
          obj['trigger_id'] = _ApiClient["default"].convertToType(data['trigger_id'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>CreateAction</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>CreateAction</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(CreateAction.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
        // ensure the json data is a string
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      if (data['description'] && !(typeof data['description'] === 'string' || data['description'] instanceof String)) {
        throw new Error("Expected the field `description` to be a primitive type in the JSON string but got " + data['description']);
      }
      // validate the optional field `email`
      if (data['email']) {
        // data not null
        _EmailAction["default"].validateJSON(data['email']);
      }
      // ensure the json data is a string
      if (data['kind'] && !(typeof data['kind'] === 'string' || data['kind'] instanceof String)) {
        throw new Error("Expected the field `kind` to be a primitive type in the JSON string but got " + data['kind']);
      }
      // ensure the json data is a string
      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      }
      // ensure the json data is a string
      if (data['organization_id'] && !(typeof data['organization_id'] === 'string' || data['organization_id'] instanceof String)) {
        throw new Error("Expected the field `organization_id` to be a primitive type in the JSON string but got " + data['organization_id']);
      }
      // validate the optional field `push_notification`
      if (data['push_notification']) {
        // data not null
        _PushAction["default"].validateJSON(data['push_notification']);
      }
      // ensure the json data is a string
      if (data['trigger_id'] && !(typeof data['trigger_id'] === 'string' || data['trigger_id'] instanceof String)) {
        throw new Error("Expected the field `trigger_id` to be a primitive type in the JSON string but got " + data['trigger_id']);
      }
      return true;
    }
  }]);
}();
CreateAction.RequiredProperties = ["kind", "name"];

/**
 * The description of the trigger
 * @member {String} description
 */
CreateAction.prototype['description'] = undefined;

/**
 * @member {module:model/EmailAction} email
 */
CreateAction.prototype['email'] = undefined;

/**
 * The kind of the action
 * @member {module:model/CreateAction.KindEnum} kind
 */
CreateAction.prototype['kind'] = undefined;

/**
 * The name of the action
 * @member {String} name
 */
CreateAction.prototype['name'] = undefined;

/**
 * Id of the organization the trigger belongs to
 * @member {String} organization_id
 */
CreateAction.prototype['organization_id'] = undefined;

/**
 * @member {module:model/PushAction} push_notification
 */
CreateAction.prototype['push_notification'] = undefined;

/**
 * Id of the trigger the action is associated to
 * @member {String} trigger_id
 */
CreateAction.prototype['trigger_id'] = undefined;

/**
 * Allowed values for the <code>kind</code> property.
 * @enum {String}
 * @readonly
 */
CreateAction['KindEnum'] = {
  /**
   * value: "NOTIFY-EMAIL"
   * @const
   */
  "EMAIL": "NOTIFY-EMAIL",
  /**
   * value: "NOTIFY-PUSH"
   * @const
   */
  "PUSH": "NOTIFY-PUSH"
};
var _default = exports["default"] = CreateAction;
