// @arduino/arduino-iot-client/dist/model/DeviceStatusSource.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(r, e) { var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (!t) { if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) { t && (r = t); var _n = 0, F = function F() {}; return { s: F, n: function n() { return _n >= r.length ? { done: !0 } : { done: !1, value: r[_n++] }; }, e: function e(r) { throw r; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var o, a = !0, u = !1; return { s: function s() { t = t.call(r); }, n: function n() { var r = t.next(); return a = r.done, r; }, e: function e(r) { u = !0, o = r; }, f: function f() { try { a || null == t["return"] || t["return"](); } finally { if (u) throw o; } } }; }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Arduino IoT Cloud API
 * Provides a set of endpoints to manage Arduino IoT Cloud **Devices**, **Things**, **Properties** and **Timeseries**. This API can be called just with any HTTP Client, or using one of these clients:  * [Javascript NPM package](https://www.npmjs.com/package/@arduino/arduino-iot-client)  * [Python PYPI Package](https://pypi.org/project/arduino-iot-client/)  * [Golang Module](https://github.com/arduino/iot-client-go)
 *
 * The version of the OpenAPI document: 2.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The DeviceStatusSource model module.
 * @module model/DeviceStatusSource
 * @version 3.0.0
 */
var DeviceStatusSource = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DeviceStatusSource</code>.
   * @alias module:model/DeviceStatusSource
   * @param criteria {module:model/DeviceStatusSource.CriteriaEnum} The matching criteria of the trigger, this allows to interpret device_ids as an inclusion or exclusion list
   */
  function DeviceStatusSource(criteria) {
    _classCallCheck(this, DeviceStatusSource);
    DeviceStatusSource.initialize(this, criteria);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(DeviceStatusSource, null, [{
    key: "initialize",
    value: function initialize(obj, criteria) {
      obj['criteria'] = criteria;
    }

    /**
     * Constructs a <code>DeviceStatusSource</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DeviceStatusSource} obj Optional instance to populate.
     * @return {module:model/DeviceStatusSource} The populated <code>DeviceStatusSource</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DeviceStatusSource();
        if (data.hasOwnProperty('criteria')) {
          obj['criteria'] = _ApiClient["default"].convertToType(data['criteria'], 'String');
        }
        if (data.hasOwnProperty('device_ids')) {
          obj['device_ids'] = _ApiClient["default"].convertToType(data['device_ids'], ['String']);
        }
        if (data.hasOwnProperty('grace_period_offline')) {
          obj['grace_period_offline'] = _ApiClient["default"].convertToType(data['grace_period_offline'], 'Number');
        }
        if (data.hasOwnProperty('grace_period_online')) {
          obj['grace_period_online'] = _ApiClient["default"].convertToType(data['grace_period_online'], 'Number');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>DeviceStatusSource</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>DeviceStatusSource</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(DeviceStatusSource.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
        // ensure the json data is a string
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      if (data['criteria'] && !(typeof data['criteria'] === 'string' || data['criteria'] instanceof String)) {
        throw new Error("Expected the field `criteria` to be a primitive type in the JSON string but got " + data['criteria']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['device_ids'])) {
        throw new Error("Expected the field `device_ids` to be an array in the JSON data but got " + data['device_ids']);
      }
      return true;
    }
  }]);
}();
DeviceStatusSource.RequiredProperties = ["criteria"];

/**
 * The matching criteria of the trigger, this allows to interpret device_ids as an inclusion or exclusion list
 * @member {module:model/DeviceStatusSource.CriteriaEnum} criteria
 */
DeviceStatusSource.prototype['criteria'] = undefined;

/**
 * A list of device IDs to be included in or excluded from the trigger (see criteria). Mutually exclusive with property_id.
 * @member {Array.<String>} device_ids
 */
DeviceStatusSource.prototype['device_ids'] = undefined;

/**
 * Amount of seconds the trigger will wait before the device will be considered disconnected (requires 'device_id')
 * @member {Number} grace_period_offline
 */
DeviceStatusSource.prototype['grace_period_offline'] = undefined;

/**
 * Amount of seconds the trigger will wait before the device will be considered connected (requires 'device_id')
 * @member {Number} grace_period_online
 */
DeviceStatusSource.prototype['grace_period_online'] = undefined;

/**
 * Allowed values for the <code>criteria</code> property.
 * @enum {String}
 * @readonly
 */
DeviceStatusSource['CriteriaEnum'] = {
  /**
   * value: "INCLUDE"
   * @const
   */
  "INCLUDE": "INCLUDE",
  /**
   * value: "EXCLUDE"
   * @const
   */
  "EXCLUDE": "EXCLUDE"
};
var _default = exports["default"] = DeviceStatusSource;
