// @arduino/arduino-iot-client/dist/model/PushAction.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _BodyExpression = _interopRequireDefault(require("./BodyExpression"));
var _PushDeliveryOpts = _interopRequireDefault(require("./PushDeliveryOpts"));
var _TitleExpression = _interopRequireDefault(require("./TitleExpression"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(r, e) { var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (!t) { if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) { t && (r = t); var _n = 0, F = function F() {}; return { s: F, n: function n() { return _n >= r.length ? { done: !0 } : { done: !1, value: r[_n++] }; }, e: function e(r) { throw r; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var o, a = !0, u = !1; return { s: function s() { t = t.call(r); }, n: function n() { var r = t.next(); return a = r.done, r; }, e: function e(r) { u = !0, o = r; }, f: function f() { try { a || null == t["return"] || t["return"](); } finally { if (u) throw o; } } }; }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Arduino IoT Cloud API
 * Provides a set of endpoints to manage Arduino IoT Cloud **Devices**, **Things**, **Properties** and **Timeseries**. This API can be called just with any HTTP Client, or using one of these clients:  * [Javascript NPM package](https://www.npmjs.com/package/@arduino/arduino-iot-client)  * [Python PYPI Package](https://pypi.org/project/arduino-iot-client/)  * [Golang Module](https://github.com/arduino/iot-client-go)
 *
 * The version of the OpenAPI document: 2.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The PushAction model module.
 * @module model/PushAction
 * @version 3.0.0
 */
var PushAction = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PushAction</code>.
   * @alias module:model/PushAction
   * @param body {module:model/BodyExpression} 
   * @param delivery {module:model/PushDeliveryOpts} 
   * @param title {module:model/TitleExpression} 
   */
  function PushAction(body, delivery, title) {
    _classCallCheck(this, PushAction);
    PushAction.initialize(this, body, delivery, title);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(PushAction, null, [{
    key: "initialize",
    value: function initialize(obj, body, delivery, title) {
      obj['body'] = body;
      obj['delivery'] = delivery;
      obj['title'] = title;
    }

    /**
     * Constructs a <code>PushAction</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PushAction} obj Optional instance to populate.
     * @return {module:model/PushAction} The populated <code>PushAction</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PushAction();
        if (data.hasOwnProperty('body')) {
          obj['body'] = _BodyExpression["default"].constructFromObject(data['body']);
        }
        if (data.hasOwnProperty('delivery')) {
          obj['delivery'] = _PushDeliveryOpts["default"].constructFromObject(data['delivery']);
        }
        if (data.hasOwnProperty('title')) {
          obj['title'] = _TitleExpression["default"].constructFromObject(data['title']);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>PushAction</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>PushAction</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(PushAction.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
        // validate the optional field `body`
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      if (data['body']) {
        // data not null
        _BodyExpression["default"].validateJSON(data['body']);
      }
      // validate the optional field `delivery`
      if (data['delivery']) {
        // data not null
        _PushDeliveryOpts["default"].validateJSON(data['delivery']);
      }
      // validate the optional field `title`
      if (data['title']) {
        // data not null
        _TitleExpression["default"].validateJSON(data['title']);
      }
      return true;
    }
  }]);
}();
PushAction.RequiredProperties = ["body", "delivery", "title"];

/**
 * @member {module:model/BodyExpression} body
 */
PushAction.prototype['body'] = undefined;

/**
 * @member {module:model/PushDeliveryOpts} delivery
 */
PushAction.prototype['delivery'] = undefined;

/**
 * @member {module:model/TitleExpression} title
 */
PushAction.prototype['title'] = undefined;
var _default = exports["default"] = PushAction;
