// @arduino/arduino-iot-client/dist/model/ThingUpdate.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _Property = _interopRequireDefault(require("./Property"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(r, e) { var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (!t) { if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) { t && (r = t); var _n = 0, F = function F() {}; return { s: F, n: function n() { return _n >= r.length ? { done: !0 } : { done: !1, value: r[_n++] }; }, e: function e(r) { throw r; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var o, a = !0, u = !1; return { s: function s() { t = t.call(r); }, n: function n() { var r = t.next(); return a = r.done, r; }, e: function e(r) { u = !0, o = r; }, f: function f() { try { a || null == t["return"] || t["return"](); } finally { if (u) throw o; } } }; }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Arduino IoT Cloud API
 * Provides a set of endpoints to manage Arduino IoT Cloud **Devices**, **Things**, **Properties** and **Timeseries**. This API can be called just with any HTTP Client, or using one of these clients:  * [Javascript NPM package](https://www.npmjs.com/package/@arduino/arduino-iot-client)  * [Python PYPI Package](https://pypi.org/project/arduino-iot-client/)  * [Golang Module](https://github.com/arduino/iot-client-go)
 *
 * The version of the OpenAPI document: 2.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The ThingUpdate model module.
 * @module model/ThingUpdate
 * @version 3.0.0
 */
var ThingUpdate = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ThingUpdate</code>.
   * Payload to update an existing thing
   * @alias module:model/ThingUpdate
   */
  function ThingUpdate() {
    _classCallCheck(this, ThingUpdate);
    ThingUpdate.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(ThingUpdate, null, [{
    key: "initialize",
    value: function initialize(obj) {
      obj['soft_deleted'] = false;
    }

    /**
     * Constructs a <code>ThingUpdate</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ThingUpdate} obj Optional instance to populate.
     * @return {module:model/ThingUpdate} The populated <code>ThingUpdate</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ThingUpdate();
        if (data.hasOwnProperty('assistant')) {
          obj['assistant'] = _ApiClient["default"].convertToType(data['assistant'], 'String');
        }
        if (data.hasOwnProperty('device_id')) {
          obj['device_id'] = _ApiClient["default"].convertToType(data['device_id'], 'String');
        }
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('properties')) {
          obj['properties'] = _ApiClient["default"].convertToType(data['properties'], [_Property["default"]]);
        }
        if (data.hasOwnProperty('soft_deleted')) {
          obj['soft_deleted'] = _ApiClient["default"].convertToType(data['soft_deleted'], 'Boolean');
        }
        if (data.hasOwnProperty('timezone')) {
          obj['timezone'] = _ApiClient["default"].convertToType(data['timezone'], 'String');
        }
        if (data.hasOwnProperty('webhook_active')) {
          obj['webhook_active'] = _ApiClient["default"].convertToType(data['webhook_active'], 'Boolean');
        }
        if (data.hasOwnProperty('webhook_uri')) {
          obj['webhook_uri'] = _ApiClient["default"].convertToType(data['webhook_uri'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>ThingUpdate</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ThingUpdate</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['assistant'] && !(typeof data['assistant'] === 'string' || data['assistant'] instanceof String)) {
        throw new Error("Expected the field `assistant` to be a primitive type in the JSON string but got " + data['assistant']);
      }
      // ensure the json data is a string
      if (data['device_id'] && !(typeof data['device_id'] === 'string' || data['device_id'] instanceof String)) {
        throw new Error("Expected the field `device_id` to be a primitive type in the JSON string but got " + data['device_id']);
      }
      // ensure the json data is a string
      if (data['id'] && !(typeof data['id'] === 'string' || data['id'] instanceof String)) {
        throw new Error("Expected the field `id` to be a primitive type in the JSON string but got " + data['id']);
      }
      // ensure the json data is a string
      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      }
      if (data['properties']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['properties'])) {
          throw new Error("Expected the field `properties` to be an array in the JSON data but got " + data['properties']);
        }
        // validate the optional field `properties` (array)
        var _iterator = _createForOfIteratorHelper(data['properties']),
          _step;
        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;
            _Property["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }
        ;
      }
      // ensure the json data is a string
      if (data['timezone'] && !(typeof data['timezone'] === 'string' || data['timezone'] instanceof String)) {
        throw new Error("Expected the field `timezone` to be a primitive type in the JSON string but got " + data['timezone']);
      }
      // ensure the json data is a string
      if (data['webhook_uri'] && !(typeof data['webhook_uri'] === 'string' || data['webhook_uri'] instanceof String)) {
        throw new Error("Expected the field `webhook_uri` to be a primitive type in the JSON string but got " + data['webhook_uri']);
      }
      return true;
    }
  }]);
}();
/**
 * The kind of voice assistant the thing is connected to, it can be ALEXA | GOOGLE | NONE
 * @member {module:model/ThingUpdate.AssistantEnum} assistant
 */
ThingUpdate.prototype['assistant'] = undefined;

/**
 * The arn of the associated device
 * @member {String} device_id
 */
ThingUpdate.prototype['device_id'] = undefined;

/**
 * The id of the thing
 * @member {String} id
 */
ThingUpdate.prototype['id'] = undefined;

/**
 * The friendly name of the thing
 * @member {String} name
 */
ThingUpdate.prototype['name'] = undefined;

/**
 * The properties of the thing
 * @member {Array.<module:model/Property>} properties
 */
ThingUpdate.prototype['properties'] = undefined;

/**
 * If false, restore the thing from the soft deletion
 * @member {Boolean} soft_deleted
 * @default false
 */
ThingUpdate.prototype['soft_deleted'] = false;

/**
 * A time zone name. Check /v2/timezones for a list of valid names.
 * @member {String} timezone
 */
ThingUpdate.prototype['timezone'] = undefined;

/**
 * Webhook uri
 * @member {Boolean} webhook_active
 */
ThingUpdate.prototype['webhook_active'] = undefined;

/**
 * Webhook uri
 * @member {String} webhook_uri
 */
ThingUpdate.prototype['webhook_uri'] = undefined;

/**
 * Allowed values for the <code>assistant</code> property.
 * @enum {String}
 * @readonly
 */
ThingUpdate['AssistantEnum'] = {
  /**
   * value: "ALEXA"
   * @const
   */
  "ALEXA": "ALEXA",
  /**
   * value: "GOOGLE"
   * @const
   */
  "GOOGLE": "GOOGLE",
  /**
   * value: "NONE"
   * @const
   */
  "NONE": "NONE"
};
var _default = exports["default"] = ThingUpdate;
