// @arduino/arduino-iot-client/dist/model/Trigger.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _CreateAction = _interopRequireDefault(require("./CreateAction"));
var _DeviceStatusSource = _interopRequireDefault(require("./DeviceStatusSource"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(r, e) { var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (!t) { if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) { t && (r = t); var _n = 0, F = function F() {}; return { s: F, n: function n() { return _n >= r.length ? { done: !0 } : { done: !1, value: r[_n++] }; }, e: function e(r) { throw r; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var o, a = !0, u = !1; return { s: function s() { t = t.call(r); }, n: function n() { var r = t.next(); return a = r.done, r; }, e: function e(r) { u = !0, o = r; }, f: function f() { try { a || null == t["return"] || t["return"](); } finally { if (u) throw o; } } }; }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Arduino IoT Cloud API
 * Provides a set of endpoints to manage Arduino IoT Cloud **Devices**, **Things**, **Properties** and **Timeseries**. This API can be called just with any HTTP Client, or using one of these clients:  * [Javascript NPM package](https://www.npmjs.com/package/@arduino/arduino-iot-client)  * [Python PYPI Package](https://pypi.org/project/arduino-iot-client/)  * [Golang Module](https://github.com/arduino/iot-client-go)
 *
 * The version of the OpenAPI document: 2.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The Trigger model module.
 * @module model/Trigger
 * @version 3.0.0
 */
var Trigger = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Trigger</code>.
   * @alias module:model/Trigger
   */
  function Trigger() {
    _classCallCheck(this, Trigger);
    Trigger.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(Trigger, null, [{
    key: "initialize",
    value: function initialize(obj) {
      obj['soft_deleted'] = false;
    }

    /**
     * Constructs a <code>Trigger</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Trigger} obj Optional instance to populate.
     * @return {module:model/Trigger} The populated <code>Trigger</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Trigger();
        if (data.hasOwnProperty('actions')) {
          obj['actions'] = _ApiClient["default"].convertToType(data['actions'], [_CreateAction["default"]]);
        }
        if (data.hasOwnProperty('active')) {
          obj['active'] = _ApiClient["default"].convertToType(data['active'], 'Boolean');
        }
        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }
        if (data.hasOwnProperty('device_status_source')) {
          obj['device_status_source'] = _DeviceStatusSource["default"].constructFromObject(data['device_status_source']);
        }
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('property_id')) {
          obj['property_id'] = _ApiClient["default"].convertToType(data['property_id'], 'String');
        }
        if (data.hasOwnProperty('soft_deleted')) {
          obj['soft_deleted'] = _ApiClient["default"].convertToType(data['soft_deleted'], 'Boolean');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>Trigger</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>Trigger</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      if (data['actions']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['actions'])) {
          throw new Error("Expected the field `actions` to be an array in the JSON data but got " + data['actions']);
        }
        // validate the optional field `actions` (array)
        var _iterator = _createForOfIteratorHelper(data['actions']),
          _step;
        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;
            _CreateAction["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }
        ;
      }
      // ensure the json data is a string
      if (data['description'] && !(typeof data['description'] === 'string' || data['description'] instanceof String)) {
        throw new Error("Expected the field `description` to be a primitive type in the JSON string but got " + data['description']);
      }
      // validate the optional field `device_status_source`
      if (data['device_status_source']) {
        // data not null
        _DeviceStatusSource["default"].validateJSON(data['device_status_source']);
      }
      // ensure the json data is a string
      if (data['id'] && !(typeof data['id'] === 'string' || data['id'] instanceof String)) {
        throw new Error("Expected the field `id` to be a primitive type in the JSON string but got " + data['id']);
      }
      // ensure the json data is a string
      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      }
      // ensure the json data is a string
      if (data['property_id'] && !(typeof data['property_id'] === 'string' || data['property_id'] instanceof String)) {
        throw new Error("Expected the field `property_id` to be a primitive type in the JSON string but got " + data['property_id']);
      }
      return true;
    }
  }]);
}();
/**
 * A list of actions to be associated with the trigger
 * @member {Array.<module:model/CreateAction>} actions
 */
Trigger.prototype['actions'] = undefined;

/**
 * Is true if the trigger is enabled
 * @member {Boolean} active
 */
Trigger.prototype['active'] = undefined;

/**
 * The description of the trigger
 * @member {String} description
 */
Trigger.prototype['description'] = undefined;

/**
 * @member {module:model/DeviceStatusSource} device_status_source
 */
Trigger.prototype['device_status_source'] = undefined;

/**
 * The id of the trigger
 * @member {String} id
 */
Trigger.prototype['id'] = undefined;

/**
 * The name of the trigger
 * @member {String} name
 */
Trigger.prototype['name'] = undefined;

/**
 * Id of the property the trigger is associated to (mutually exclusive with 'device_status_source')
 * @member {String} property_id
 */
Trigger.prototype['property_id'] = undefined;

/**
 * If false, restore the thing from the soft deletion
 * @member {Boolean} soft_deleted
 * @default false
 */
Trigger.prototype['soft_deleted'] = false;
var _default = exports["default"] = Trigger;
