// @arduino/arduino-iot-client/dist/model/UpdateSketch.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Arduino IoT Cloud API
 * Provides a set of endpoints to manage Arduino IoT Cloud **Devices**, **Things**, **Properties** and **Timeseries**. This API can be called just with any HTTP Client, or using one of these clients:  * [Javascript NPM package](https://www.npmjs.com/package/@arduino/arduino-iot-client)  * [Python PYPI Package](https://pypi.org/project/arduino-iot-client/)  * [Golang Module](https://github.com/arduino/iot-client-go)
 *
 * The version of the OpenAPI document: 2.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The UpdateSketch model module.
 * @module model/UpdateSketch
 * @version 3.0.0
 */
var UpdateSketch = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateSketch</code>.
   * @alias module:model/UpdateSketch
   */
  function UpdateSketch() {
    _classCallCheck(this, UpdateSketch);
    UpdateSketch.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(UpdateSketch, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>UpdateSketch</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateSketch} obj Optional instance to populate.
     * @return {module:model/UpdateSketch} The populated <code>UpdateSketch</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateSketch();
        if (data.hasOwnProperty('sketch_version')) {
          obj['sketch_version'] = _ApiClient["default"].convertToType(data['sketch_version'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>UpdateSketch</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>UpdateSketch</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['sketch_version'] && !(typeof data['sketch_version'] === 'string' || data['sketch_version'] instanceof String)) {
        throw new Error("Expected the field `sketch_version` to be a primitive type in the JSON string but got " + data['sketch_version']);
      }
      return true;
    }
  }]);
}();
/**
 * The autogenerated sketch version
 * @member {module:model/UpdateSketch.SketchVersionEnum} sketch_version
 */
UpdateSketch.prototype['sketch_version'] = undefined;

/**
 * Allowed values for the <code>sketch_version</code> property.
 * @enum {String}
 * @readonly
 */
UpdateSketch['SketchVersionEnum'] = {
  /**
   * value: "v1"
   * @const
   */
  "v1": "v1",
  /**
   * value: "v2"
   * @const
   */
  "v2": "v2"
};
var _default = exports["default"] = UpdateSketch;
