// @arduino/arduino-iot-client/dist/model/Variable.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(r, e) { var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (!t) { if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) { t && (r = t); var _n = 0, F = function F() {}; return { s: F, n: function n() { return _n >= r.length ? { done: !0 } : { done: !1, value: r[_n++] }; }, e: function e(r) { throw r; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var o, a = !0, u = !1; return { s: function s() { t = t.call(r); }, n: function n() { var r = t.next(); return a = r.done, r; }, e: function e(r) { u = !0, o = r; }, f: function f() { try { a || null == t["return"] || t["return"](); } finally { if (u) throw o; } } }; }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Arduino IoT Cloud API
 * Provides a set of endpoints to manage Arduino IoT Cloud **Devices**, **Things**, **Properties** and **Timeseries**. This API can be called just with any HTTP Client, or using one of these clients:  * [Javascript NPM package](https://www.npmjs.com/package/@arduino/arduino-iot-client)  * [Python PYPI Package](https://pypi.org/project/arduino-iot-client/)  * [Golang Module](https://github.com/arduino/iot-client-go)
 *
 * The version of the OpenAPI document: 2.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The Variable model module.
 * @module model/Variable
 * @version 3.0.0
 */
var Variable = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Variable</code>.
   * @alias module:model/Variable
   * @param attribute {module:model/Variable.AttributeEnum} The template expression that extracts the value from the respective entity
   * @param entity {module:model/Variable.EntityEnum} Type of the entity being referenced
   * @param placeholder {String} Name of the variable as referenced by the expression
   */
  function Variable(attribute, entity, placeholder) {
    _classCallCheck(this, Variable);
    Variable.initialize(this, attribute, entity, placeholder);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(Variable, null, [{
    key: "initialize",
    value: function initialize(obj, attribute, entity, placeholder) {
      obj['attribute'] = attribute;
      obj['entity'] = entity;
      obj['placeholder'] = placeholder;
    }

    /**
     * Constructs a <code>Variable</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Variable} obj Optional instance to populate.
     * @return {module:model/Variable} The populated <code>Variable</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Variable();
        if (data.hasOwnProperty('attribute')) {
          obj['attribute'] = _ApiClient["default"].convertToType(data['attribute'], 'String');
        }
        if (data.hasOwnProperty('entity')) {
          obj['entity'] = _ApiClient["default"].convertToType(data['entity'], 'String');
        }
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
        if (data.hasOwnProperty('placeholder')) {
          obj['placeholder'] = _ApiClient["default"].convertToType(data['placeholder'], 'String');
        }
        if (data.hasOwnProperty('property_id')) {
          obj['property_id'] = _ApiClient["default"].convertToType(data['property_id'], 'String');
        }
        if (data.hasOwnProperty('thing_id')) {
          obj['thing_id'] = _ApiClient["default"].convertToType(data['thing_id'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>Variable</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>Variable</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(Variable.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
        // ensure the json data is a string
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      if (data['attribute'] && !(typeof data['attribute'] === 'string' || data['attribute'] instanceof String)) {
        throw new Error("Expected the field `attribute` to be a primitive type in the JSON string but got " + data['attribute']);
      }
      // ensure the json data is a string
      if (data['entity'] && !(typeof data['entity'] === 'string' || data['entity'] instanceof String)) {
        throw new Error("Expected the field `entity` to be a primitive type in the JSON string but got " + data['entity']);
      }
      // ensure the json data is a string
      if (data['id'] && !(typeof data['id'] === 'string' || data['id'] instanceof String)) {
        throw new Error("Expected the field `id` to be a primitive type in the JSON string but got " + data['id']);
      }
      // ensure the json data is a string
      if (data['placeholder'] && !(typeof data['placeholder'] === 'string' || data['placeholder'] instanceof String)) {
        throw new Error("Expected the field `placeholder` to be a primitive type in the JSON string but got " + data['placeholder']);
      }
      // ensure the json data is a string
      if (data['property_id'] && !(typeof data['property_id'] === 'string' || data['property_id'] instanceof String)) {
        throw new Error("Expected the field `property_id` to be a primitive type in the JSON string but got " + data['property_id']);
      }
      // ensure the json data is a string
      if (data['thing_id'] && !(typeof data['thing_id'] === 'string' || data['thing_id'] instanceof String)) {
        throw new Error("Expected the field `thing_id` to be a primitive type in the JSON string but got " + data['thing_id']);
      }
      return true;
    }
  }]);
}();
Variable.RequiredProperties = ["attribute", "entity", "placeholder"];

/**
 * The template expression that extracts the value from the respective entity
 * @member {module:model/Variable.AttributeEnum} attribute
 */
Variable.prototype['attribute'] = undefined;

/**
 * Type of the entity being referenced
 * @member {module:model/Variable.EntityEnum} entity
 */
Variable.prototype['entity'] = undefined;

/**
 * The ID of the referenced entity
 * @member {String} id
 */
Variable.prototype['id'] = undefined;

/**
 * Name of the variable as referenced by the expression
 * @member {String} placeholder
 */
Variable.prototype['placeholder'] = undefined;

/**
 * The ID of the property referenced entity
 * @member {String} property_id
 */
Variable.prototype['property_id'] = undefined;

/**
 * The ID of the thing referenced entity
 * @member {String} thing_id
 */
Variable.prototype['thing_id'] = undefined;

/**
 * Allowed values for the <code>attribute</code> property.
 * @enum {String}
 * @readonly
 */
Variable['AttributeEnum'] = {
  /**
   * value: "ID"
   * @const
   */
  "ID": "ID",
  /**
   * value: "NAME"
   * @const
   */
  "NAME": "NAME",
  /**
   * value: "VALUE"
   * @const
   */
  "VALUE": "VALUE",
  /**
   * value: "TIMESTAMP"
   * @const
   */
  "TIMESTAMP": "TIMESTAMP",
  /**
   * value: "USERNAME"
   * @const
   */
  "USERNAME": "USERNAME",
  /**
   * value: "EMAIL"
   * @const
   */
  "EMAIL": "EMAIL",
  /**
   * value: "STATUS"
   * @const
   */
  "STATUS": "STATUS"
};

/**
 * Allowed values for the <code>entity</code> property.
 * @enum {String}
 * @readonly
 */
Variable['EntityEnum'] = {
  /**
   * value: "RECIPIENT"
   * @const
   */
  "RECIPIENT": "RECIPIENT",
  /**
   * value: "USER"
   * @const
   */
  "USER": "USER",
  /**
   * value: "DEVICE"
   * @const
   */
  "DEVICE": "DEVICE",
  /**
   * value: "THING"
   * @const
   */
  "THING": "THING",
  /**
   * value: "PROPERTY"
   * @const
   */
  "PROPERTY": "PROPERTY",
  /**
   * value: "TRIGGERING_DEVICE"
   * @const
   */
  "TRIGGERING_DEVICE": "TRIGGERING_DEVICE"
};
var _default = exports["default"] = Variable;
