// @arduino/arduino-iot-client/dist/api/DevicesV2PassApi.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ArduinoDevicev2Pass = _interopRequireDefault(require("../model/ArduinoDevicev2Pass"));
var _CheckDevicesV2PassPayload = _interopRequireDefault(require("../model/CheckDevicesV2PassPayload"));
var _Devicev2Pass = _interopRequireDefault(require("../model/Devicev2Pass"));
var _Error = _interopRequireDefault(require("../model/Error"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Arduino IoT Cloud API
 * Provides a set of endpoints to manage Arduino IoT Cloud **Devices**, **Things**, **Properties** and **Timeseries**. This API can be called just with any HTTP Client, or using one of these clients:  * [Javascript NPM package](https://www.npmjs.com/package/@arduino/arduino-iot-client)  * [Python PYPI Package](https://pypi.org/project/arduino-iot-client/)  * [Golang Module](https://github.com/arduino/iot-client-go)
 *
 * The version of the OpenAPI document: 2.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
* DevicesV2Pass service.
* @module api/DevicesV2PassApi
* @version 3.0.0
*/
var DevicesV2PassApi = exports["default"] = /*#__PURE__*/function () {
  /**
  * Constructs a new DevicesV2PassApi. 
  * @alias module:api/DevicesV2PassApi
  * @class
  * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
  * default to {@link module:ApiClient#instance} if unspecified.
  */
  function DevicesV2PassApi(apiClient) {
    _classCallCheck(this, DevicesV2PassApi);
    this.apiClient = apiClient || _ApiClient["default"].instance;
  }

  /**
   * check devices_v2_pass
   * Check if the password matches.
   * @param {String} id The id of the device
   * @param {module:model/CheckDevicesV2PassPayload} checkDevicesV2PassPayload 
   * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing HTTP response
   */
  return _createClass(DevicesV2PassApi, [{
    key: "devicesV2PassCheckWithHttpInfo",
    value: function devicesV2PassCheckWithHttpInfo(id, checkDevicesV2PassPayload) {
      var postBody = checkDevicesV2PassPayload;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new _Error["default"]("Missing the required parameter 'id' when calling devicesV2PassCheck");
      }
      // verify the required parameter 'checkDevicesV2PassPayload' is set
      if (checkDevicesV2PassPayload === undefined || checkDevicesV2PassPayload === null) {
        throw new _Error["default"]("Missing the required parameter 'checkDevicesV2PassPayload' when calling devicesV2PassCheck");
      }
      var pathParams = {
        'id': id
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['oauth2'];
      var contentTypes = ['application/json'];
      var accepts = ['application/vnd.goa.error+json', 'text/plain'];
      var returnType = null;
      return this.apiClient.callApi('/iot/v2/devices/{id}/pass', 'POST', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * check devices_v2_pass
     * Check if the password matches.
     * @param {String} id The id of the device
     * @param {module:model/CheckDevicesV2PassPayload} checkDevicesV2PassPayload 
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}
     */
  }, {
    key: "devicesV2PassCheck",
    value: function devicesV2PassCheck(id, checkDevicesV2PassPayload) {
      return this.devicesV2PassCheckWithHttpInfo(id, checkDevicesV2PassPayload).then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * delete devices_v2_pass
     * Removes the password for the device.
     * @param {String} id The id of the device
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing HTTP response
     */
  }, {
    key: "devicesV2PassDeleteWithHttpInfo",
    value: function devicesV2PassDeleteWithHttpInfo(id) {
      var postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new _Error["default"]("Missing the required parameter 'id' when calling devicesV2PassDelete");
      }
      var pathParams = {
        'id': id
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['oauth2'];
      var contentTypes = [];
      var accepts = ['application/vnd.goa.error+json', 'text/plain'];
      var returnType = null;
      return this.apiClient.callApi('/iot/v2/devices/{id}/pass', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * delete devices_v2_pass
     * Removes the password for the device.
     * @param {String} id The id of the device
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}
     */
  }, {
    key: "devicesV2PassDelete",
    value: function devicesV2PassDelete(id) {
      return this.devicesV2PassDeleteWithHttpInfo(id).then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * get devices_v2_pass
     * Returns whether the password for this device is set or not. It doesn't return the password.
     * @param {String} id The id of the device
     * @param {Object} opts Optional parameters
     * @param {Boolean} [suggestedPassword = false)] If true, return a suggested password
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:model/ArduinoDevicev2Pass} and HTTP response
     */
  }, {
    key: "devicesV2PassGetWithHttpInfo",
    value: function devicesV2PassGetWithHttpInfo(id, opts) {
      opts = opts || {};
      var postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new _Error["default"]("Missing the required parameter 'id' when calling devicesV2PassGet");
      }
      var pathParams = {
        'id': id
      };
      var queryParams = {
        'suggested_password': opts['suggestedPassword']
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['oauth2'];
      var contentTypes = [];
      var accepts = ['application/vnd.arduino.devicev2.pass+json', 'application/vnd.goa.error+json'];
      var returnType = _ArduinoDevicev2Pass["default"];
      return this.apiClient.callApi('/iot/v2/devices/{id}/pass', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * get devices_v2_pass
     * Returns whether the password for this device is set or not. It doesn't return the password.
     * @param {String} id The id of the device
     * @param {Object} opts Optional parameters
     * @param {Boolean} opts.suggestedPassword If true, return a suggested password (default to false)
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:model/ArduinoDevicev2Pass}
     */
  }, {
    key: "devicesV2PassGet",
    value: function devicesV2PassGet(id, opts) {
      return this.devicesV2PassGetWithHttpInfo(id, opts).then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * set devices_v2_pass
     * Sets the password for the device. It can never be read back.
     * @param {String} id The id of the device
     * @param {module:model/Devicev2Pass} devicev2Pass 
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:model/ArduinoDevicev2Pass} and HTTP response
     */
  }, {
    key: "devicesV2PassSetWithHttpInfo",
    value: function devicesV2PassSetWithHttpInfo(id, devicev2Pass) {
      var postBody = devicev2Pass;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new _Error["default"]("Missing the required parameter 'id' when calling devicesV2PassSet");
      }
      // verify the required parameter 'devicev2Pass' is set
      if (devicev2Pass === undefined || devicev2Pass === null) {
        throw new _Error["default"]("Missing the required parameter 'devicev2Pass' when calling devicesV2PassSet");
      }
      var pathParams = {
        'id': id
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['oauth2'];
      var contentTypes = ['application/json'];
      var accepts = ['application/vnd.arduino.devicev2.pass+json', 'application/vnd.goa.error+json'];
      var returnType = _ArduinoDevicev2Pass["default"];
      return this.apiClient.callApi('/iot/v2/devices/{id}/pass', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * set devices_v2_pass
     * Sets the password for the device. It can never be read back.
     * @param {String} id The id of the device
     * @param {module:model/Devicev2Pass} devicev2Pass 
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:model/ArduinoDevicev2Pass}
     */
  }, {
    key: "devicesV2PassSet",
    value: function devicesV2PassSet(id, devicev2Pass) {
      return this.devicesV2PassSetWithHttpInfo(id, devicev2Pass).then(function (response_and_data) {
        return response_and_data.data;
      });
    }
  }]);
}();
