// @arduino/arduino-iot-client/dist/api/TemplatesApi.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ArduinoTemplate = _interopRequireDefault(require("../model/ArduinoTemplate"));
var _Error = _interopRequireDefault(require("../model/Error"));
var _Template = _interopRequireDefault(require("../model/Template"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Arduino IoT Cloud API
 * Provides a set of endpoints to manage Arduino IoT Cloud **Devices**, **Things**, **Properties** and **Timeseries**. This API can be called just with any HTTP Client, or using one of these clients:  * [Javascript NPM package](https://www.npmjs.com/package/@arduino/arduino-iot-client)  * [Python PYPI Package](https://pypi.org/project/arduino-iot-client/)  * [Golang Module](https://github.com/arduino/iot-client-go)
 *
 * The version of the OpenAPI document: 2.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
* Templates service.
* @module api/TemplatesApi
* @version 3.0.0
*/
var TemplatesApi = exports["default"] = /*#__PURE__*/function () {
  /**
  * Constructs a new TemplatesApi. 
  * @alias module:api/TemplatesApi
  * @class
  * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
  * default to {@link module:ApiClient#instance} if unspecified.
  */
  function TemplatesApi(apiClient) {
    _classCallCheck(this, TemplatesApi);
    this.apiClient = apiClient || _ApiClient["default"].instance;
  }

  /**
   * apply templates
   * Apply an existing cloud template and generate all the needed resources
   * @param {module:model/Template} template TemplatePayload describes the needed attribute to apply a template
   * @param {Object} opts Optional parameters
   * @param {String} [xOrganization] Organization space identifer (optional)
   * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:model/ArduinoTemplate} and HTTP response
   */
  return _createClass(TemplatesApi, [{
    key: "templatesApplyWithHttpInfo",
    value: function templatesApplyWithHttpInfo(template, opts) {
      opts = opts || {};
      var postBody = template;
      // verify the required parameter 'template' is set
      if (template === undefined || template === null) {
        throw new _Error["default"]("Missing the required parameter 'template' when calling templatesApply");
      }
      var pathParams = {};
      var queryParams = {};
      var headerParams = {
        'X-Organization': opts['xOrganization']
      };
      var formParams = {};
      var authNames = ['oauth2'];
      var contentTypes = ['application/json'];
      var accepts = ['application/vnd.arduino.template+json', 'application/vnd.goa.error+json'];
      var returnType = _ArduinoTemplate["default"];
      return this.apiClient.callApi('/iot/v1/templates', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * apply templates
     * Apply an existing cloud template and generate all the needed resources
     * @param {module:model/Template} template TemplatePayload describes the needed attribute to apply a template
     * @param {Object} opts Optional parameters
     * @param {String} opts.xOrganization Organization space identifer (optional)
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:model/ArduinoTemplate}
     */
  }, {
    key: "templatesApply",
    value: function templatesApply(template, opts) {
      return this.templatesApplyWithHttpInfo(template, opts).then(function (response_and_data) {
        return response_and_data.data;
      });
    }
  }]);
}();
