// @arduino/arduino-iot-client/dist/api/ThingsV2TagsApi.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ArduinoTags = _interopRequireDefault(require("../model/ArduinoTags"));
var _Error = _interopRequireDefault(require("../model/Error"));
var _Tag = _interopRequireDefault(require("../model/Tag"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Arduino IoT Cloud API
 * Provides a set of endpoints to manage Arduino IoT Cloud **Devices**, **Things**, **Properties** and **Timeseries**. This API can be called just with any HTTP Client, or using one of these clients:  * [Javascript NPM package](https://www.npmjs.com/package/@arduino/arduino-iot-client)  * [Python PYPI Package](https://pypi.org/project/arduino-iot-client/)  * [Golang Module](https://github.com/arduino/iot-client-go)
 *
 * The version of the OpenAPI document: 2.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
* ThingsV2Tags service.
* @module api/ThingsV2TagsApi
* @version 3.0.0
*/
var ThingsV2TagsApi = exports["default"] = /*#__PURE__*/function () {
  /**
  * Constructs a new ThingsV2TagsApi. 
  * @alias module:api/ThingsV2TagsApi
  * @class
  * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
  * default to {@link module:ApiClient#instance} if unspecified.
  */
  function ThingsV2TagsApi(apiClient) {
    _classCallCheck(this, ThingsV2TagsApi);
    this.apiClient = apiClient || _ApiClient["default"].instance;
  }

  /**
   * delete things_v2_tags
   * Delete a tag associated to the thing given its key.
   * @param {String} id The id of the thing
   * @param {String} key The key of the tag
   * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing HTTP response
   */
  return _createClass(ThingsV2TagsApi, [{
    key: "thingsV2TagsDeleteWithHttpInfo",
    value: function thingsV2TagsDeleteWithHttpInfo(id, key) {
      var postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new _Error["default"]("Missing the required parameter 'id' when calling thingsV2TagsDelete");
      }
      // verify the required parameter 'key' is set
      if (key === undefined || key === null) {
        throw new _Error["default"]("Missing the required parameter 'key' when calling thingsV2TagsDelete");
      }
      var pathParams = {
        'id': id,
        'key': key
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['oauth2'];
      var contentTypes = [];
      var accepts = ['application/vnd.goa.error+json', 'text/plain'];
      var returnType = null;
      return this.apiClient.callApi('/iot/v2/things/{id}/tags/{key}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * delete things_v2_tags
     * Delete a tag associated to the thing given its key.
     * @param {String} id The id of the thing
     * @param {String} key The key of the tag
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}
     */
  }, {
    key: "thingsV2TagsDelete",
    value: function thingsV2TagsDelete(id, key) {
      return this.thingsV2TagsDeleteWithHttpInfo(id, key).then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * list things_v2_tags
     * List tags associated to the thing.
     * @param {String} id The id of the thing
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:model/ArduinoTags} and HTTP response
     */
  }, {
    key: "thingsV2TagsListWithHttpInfo",
    value: function thingsV2TagsListWithHttpInfo(id) {
      var postBody = null;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new _Error["default"]("Missing the required parameter 'id' when calling thingsV2TagsList");
      }
      var pathParams = {
        'id': id
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['oauth2'];
      var contentTypes = [];
      var accepts = ['application/vnd.arduino.tags+json', 'application/vnd.goa.error+json'];
      var returnType = _ArduinoTags["default"];
      return this.apiClient.callApi('/iot/v2/things/{id}/tags', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * list things_v2_tags
     * List tags associated to the thing.
     * @param {String} id The id of the thing
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:model/ArduinoTags}
     */
  }, {
    key: "thingsV2TagsList",
    value: function thingsV2TagsList(id) {
      return this.thingsV2TagsListWithHttpInfo(id).then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * upsert things_v2_tags
     * Creates or updates a tag associated to the thing.
     * @param {String} id The id of the thing
     * @param {module:model/Tag} tag 
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing HTTP response
     */
  }, {
    key: "thingsV2TagsUpsertWithHttpInfo",
    value: function thingsV2TagsUpsertWithHttpInfo(id, tag) {
      var postBody = tag;
      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new _Error["default"]("Missing the required parameter 'id' when calling thingsV2TagsUpsert");
      }
      // verify the required parameter 'tag' is set
      if (tag === undefined || tag === null) {
        throw new _Error["default"]("Missing the required parameter 'tag' when calling thingsV2TagsUpsert");
      }
      var pathParams = {
        'id': id
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['oauth2'];
      var contentTypes = ['application/json'];
      var accepts = ['application/vnd.goa.error+json', 'text/plain'];
      var returnType = null;
      return this.apiClient.callApi('/iot/v2/things/{id}/tags', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * upsert things_v2_tags
     * Creates or updates a tag associated to the thing.
     * @param {String} id The id of the thing
     * @param {module:model/Tag} tag 
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}
     */
  }, {
    key: "thingsV2TagsUpsert",
    value: function thingsV2TagsUpsert(id, tag) {
      return this.thingsV2TagsUpsertWithHttpInfo(id, tag).then(function (response_and_data) {
        return response_and_data.data;
      });
    }
  }]);
}();
